# PROYECTO FINAL 

##### Ejecutamos tecnicas de ETL de 4 dataframe que se usaran para completar el análisis sobre los jugadores que contienen los datos específicos de la temporada (2021-2022-2023) de cada jugador (puntos, bloqueos, asistencias, salarios, entre otros).

Se importan las librerias que se consideraron para realizar el ETL en esta fase del proyecto

In [2996]:
import pandas as pd
import numpy as np

ANALISIS DE LA TEMPORADA 2021

Carga del dataframe "nba2021PerGame"

Revision del dataframe (info, conteo y nulos)

In [2997]:
df1 = pd.read_csv("nba2021_per_game.csv") 

In [2998]:
# Primeras filas del dataframe
print(df1.head(11))

                      Player Pos  Age   Tm   G  GS    MP   FG   FGA    FG%  \
0           Precious Achiuwa  PF   21  MIA  28   2  14.6  2.6   4.4  0.590   
1               Jaylen Adams  PG   24  MIL   6   0   2.8  0.2   1.3  0.125   
2               Steven Adams   C   27  NOP  27  27  28.1  3.5   5.8  0.603   
3                Bam Adebayo   C   23  MIA  26  26  33.6  7.4  12.9  0.573   
4          LaMarcus Aldridge   C   35  SAS  18  18  26.7  5.9  12.5  0.476   
5          Ty-Shon Alexander  SG   22  PHO   3   0   2.7  0.0   1.0  0.000   
6   Nickeil Alexander-Walker  SG   22  NOP  23   3  19.2  3.3   8.2  0.410   
7              Grayson Allen  SG   25  MEM  19   8  23.9  3.2   7.4  0.429   
8              Jarrett Allen   C   22  TOT  28  10  26.2  4.4   6.8  0.642   
9              Jarrett Allen   C   22  BRK  12   5  26.7  3.7   5.4  0.677   
10             Jarrett Allen   C   22  CLE  16   5  25.9  4.9   7.8  0.624   

     3P  3PA    3P%   2P   2PA    2P%   eFG%   FT  FTA    FT%  

In [2999]:
# Información general sobre el dataframe
print(df1.info())

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 497 entries, 0 to 496
Data columns (total 29 columns):
 #   Column  Non-Null Count  Dtype  
---  ------  --------------  -----  
 0   Player  497 non-null    object 
 1   Pos     497 non-null    object 
 2   Age     497 non-null    int64  
 3   Tm      497 non-null    object 
 4   G       497 non-null    int64  
 5   GS      497 non-null    int64  
 6   MP      497 non-null    float64
 7   FG      497 non-null    float64
 8   FGA     497 non-null    float64
 9   FG%     497 non-null    float64
 10  3P      497 non-null    float64
 11  3PA     497 non-null    float64
 12  3P%     497 non-null    float64
 13  2P      497 non-null    float64
 14  2PA     497 non-null    float64
 15  2P%     497 non-null    float64
 16  eFG%    497 non-null    float64
 17  FT      497 non-null    float64
 18  FTA     497 non-null    float64
 19  FT%     497 non-null    float64
 20  ORB     497 non-null    float64
 21  DRB     497 non-null    float64
 22  TR

In [3000]:
np.shape(df1)

(497, 29)

In [3001]:
#Revision Nulos 
ValoresNull1 = df1.isnull().sum() [df1.isnull().sum()>0]
print(ValoresNull1)

Series([], dtype: int64)


CAMBIAR NOMBRE DE COLUMNAS 

Se realiza modificacion de nombres de columnas para unificar criterios con el resto de las tablas 

In [3002]:
nuevos_nombres = {
    'Tm': 'Team',
    'G': 'GP'
}

#Renombrar las columnas especificadas
df1.rename(columns=nuevos_nombres, inplace=True)

print(df1.head())

              Player Pos  Age Team  GP  GS    MP   FG   FGA    FG%   3P  3PA  \
0   Precious Achiuwa  PF   21  MIA  28   2  14.6  2.6   4.4  0.590  0.0  0.0   
1       Jaylen Adams  PG   24  MIL   6   0   2.8  0.2   1.3  0.125  0.0  0.3   
2       Steven Adams   C   27  NOP  27  27  28.1  3.5   5.8  0.603  0.0  0.0   
3        Bam Adebayo   C   23  MIA  26  26  33.6  7.4  12.9  0.573  0.1  0.2   
4  LaMarcus Aldridge   C   35  SAS  18  18  26.7  5.9  12.5  0.476  1.3  3.7   

     3P%   2P   2PA    2P%   eFG%   FT  FTA    FT%  ORB  DRB  TRB  AST  STL  \
0  0.000  2.6   4.4  0.590  0.590  1.3  2.4  0.561  1.3  2.7  4.0  0.6  0.4   
1  0.000  0.2   1.0  0.167  0.125  0.0  0.0  0.000  0.0  0.5  0.5  0.3  0.0   
2  0.000  3.5   5.7  0.606  0.603  1.1  2.3  0.468  4.3  4.6  8.9  2.1  1.0   
3  0.400  7.3  12.7  0.576  0.576  5.1  6.0  0.841  1.9  7.3  9.2  5.3  1.0   
4  0.358  4.6   8.8  0.525  0.529  0.9  1.2  0.762  0.8  3.5  4.3  1.9  0.4   

   BLK  TOV   PF   PTS  
0  0.5  1.0  1.9   

In [3003]:
# Año que deseas agregar
Temporada = 2021

# Agregar una nueva columna 'Temporada' con el valor del año especificado
df1['Temp'] = Temporada

# Mostrar las primeras filas del DataFrame para verificar
print(df1.head())

              Player Pos  Age Team  GP  GS    MP   FG   FGA    FG%   3P  3PA  \
0   Precious Achiuwa  PF   21  MIA  28   2  14.6  2.6   4.4  0.590  0.0  0.0   
1       Jaylen Adams  PG   24  MIL   6   0   2.8  0.2   1.3  0.125  0.0  0.3   
2       Steven Adams   C   27  NOP  27  27  28.1  3.5   5.8  0.603  0.0  0.0   
3        Bam Adebayo   C   23  MIA  26  26  33.6  7.4  12.9  0.573  0.1  0.2   
4  LaMarcus Aldridge   C   35  SAS  18  18  26.7  5.9  12.5  0.476  1.3  3.7   

     3P%   2P   2PA    2P%   eFG%   FT  FTA    FT%  ORB  DRB  TRB  AST  STL  \
0  0.000  2.6   4.4  0.590  0.590  1.3  2.4  0.561  1.3  2.7  4.0  0.6  0.4   
1  0.000  0.2   1.0  0.167  0.125  0.0  0.0  0.000  0.0  0.5  0.5  0.3  0.0   
2  0.000  3.5   5.7  0.606  0.603  1.1  2.3  0.468  4.3  4.6  8.9  2.1  1.0   
3  0.400  7.3  12.7  0.576  0.576  5.1  6.0  0.841  1.9  7.3  9.2  5.3  1.0   
4  0.358  4.6   8.8  0.525  0.529  0.9  1.2  0.762  0.8  3.5  4.3  1.9  0.4   

   BLK  TOV   PF   PTS  Temp  
0  0.5  1.0  

In [3004]:
print(df1.head())

              Player Pos  Age Team  GP  GS    MP   FG   FGA    FG%   3P  3PA  \
0   Precious Achiuwa  PF   21  MIA  28   2  14.6  2.6   4.4  0.590  0.0  0.0   
1       Jaylen Adams  PG   24  MIL   6   0   2.8  0.2   1.3  0.125  0.0  0.3   
2       Steven Adams   C   27  NOP  27  27  28.1  3.5   5.8  0.603  0.0  0.0   
3        Bam Adebayo   C   23  MIA  26  26  33.6  7.4  12.9  0.573  0.1  0.2   
4  LaMarcus Aldridge   C   35  SAS  18  18  26.7  5.9  12.5  0.476  1.3  3.7   

     3P%   2P   2PA    2P%   eFG%   FT  FTA    FT%  ORB  DRB  TRB  AST  STL  \
0  0.000  2.6   4.4  0.590  0.590  1.3  2.4  0.561  1.3  2.7  4.0  0.6  0.4   
1  0.000  0.2   1.0  0.167  0.125  0.0  0.0  0.000  0.0  0.5  0.5  0.3  0.0   
2  0.000  3.5   5.7  0.606  0.603  1.1  2.3  0.468  4.3  4.6  8.9  2.1  1.0   
3  0.400  7.3  12.7  0.576  0.576  5.1  6.0  0.841  1.9  7.3  9.2  5.3  1.0   
4  0.358  4.6   8.8  0.525  0.529  0.9  1.2  0.762  0.8  3.5  4.3  1.9  0.4   

   BLK  TOV   PF   PTS  Temp  
0  0.5  1.0  

In [3005]:
np.shape(df1)

(497, 30)

In [3006]:
# Obtener los valores únicos de la columna team_abbreviation
valores_unicos = df1['Team'].unique()

# Mostrar los valores únicos
print(valores_unicos)

['MIA' 'MIL' 'NOP' 'SAS' 'PHO' 'MEM' 'TOT' 'BRK' 'CLE' 'ORL' 'LAL' 'POR'
 'TOR' 'CHI' 'WAS' 'UTA' 'SAC' 'CHO' 'NYK' 'DEN' 'LAC' 'GSW' 'OKC' 'MIN'
 'DET' 'DAL' 'IND' 'ATL' 'PHI' 'BOS' 'HOU']


In [3007]:
# Filtrar filas donde 'Team' es 'TOT'
df_tot = df1[df1['Team'] == 'TOT']

# Mostrar el resultado
print(df_tot)

               Player    Pos  Age Team  GP  GS    MP   FG   FGA    FG%   3P  \
8       Jarrett Allen      C   22  TOT  28  10  26.2  4.4   6.8  0.642  0.0   
181      James Harden     SG   31  TOT  24  24  37.6  7.6  15.8  0.482  3.0   
255    Rodions Kurucs  SF-PF   22  TOT  14   0   5.2  0.4   1.4  0.300  0.2   
265          Alex Len      C   27  TOT  20   6  14.6  2.4   3.8  0.627  0.3   
354    Victor Oladipo     SG   28  TOT  20  20  32.0  7.0  17.5  0.401  2.4   
382    Taurean Prince     PF   26  TOT  27  10  21.9  2.8   7.2  0.392  1.4   
403      Derrick Rose     PG   32  TOT  20   0  22.3  5.0  11.7  0.427  0.8   
425  Dennis Smith Jr.     PG   23  TOT   7   0  12.1  1.6   4.7  0.333  0.3   

     3PA    3P%   2P   2PA    2P%   eFG%   FT  FTA    FT%  ORB  DRB  TRB  \
8    0.1  0.250  4.3   6.6  0.651  0.645  3.6  4.7  0.758  2.9  6.1  9.0   
181  7.9  0.384  4.6   7.9  0.579  0.578  6.2  6.9  0.897  0.8  6.5  7.3   
255  1.0  0.214  0.2   0.4  0.500  0.375  0.1  0.1  0.500  0

In [3008]:
# Obtener los valores únicos de la columna team_abbreviation
valores_unicos = df_tot['Player'].unique()

# Mostrar los valores únicos
print(valores_unicos)

['Jarrett Allen' 'James Harden' 'Rodions Kurucs' 'Alex Len'
 'Victor Oladipo' 'Taurean Prince' 'Derrick Rose' 'Dennis Smith Jr.']


In [3009]:
# Mapeo de jugadores a equipos
player_to_team = {
    'Jarrett Allen': 'CLE',
    'James Harden': 'BKN',
    'Rodions Kurucs': 'BKN',
    'Alex Len': 'WAS',
    'Victor Oladipo': 'IND',
    'Taurean Prince': 'BKN',
    'Derrick Rose': 'DET',
    'Dennis Smith Jr.': 'NYK'
}

# Reemplazar 'TOT' por el equipo correspondiente
def replace_team(row):
    if 'TOT' in row['Team']:
        return player_to_team.get(row['Player'], row['Team'])  # Reemplaza si hay un mapeo, de lo contrario mantiene el valor original
    return row['Team']

df1['Team'] = df1.apply(replace_team, axis=1)

print(df1)

                Player Pos  Age Team  GP  GS    MP   FG   FGA    FG%   3P  \
0     Precious Achiuwa  PF   21  MIA  28   2  14.6  2.6   4.4  0.590  0.0   
1         Jaylen Adams  PG   24  MIL   6   0   2.8  0.2   1.3  0.125  0.0   
2         Steven Adams   C   27  NOP  27  27  28.1  3.5   5.8  0.603  0.0   
3          Bam Adebayo   C   23  MIA  26  26  33.6  7.4  12.9  0.573  0.1   
4    LaMarcus Aldridge   C   35  SAS  18  18  26.7  5.9  12.5  0.476  1.3   
..                 ...  ..  ...  ...  ..  ..   ...  ...   ...    ...  ...   
492       Delon Wright  SG   28  DET  28  23  29.4  3.7   8.1  0.456  0.9   
493     Thaddeus Young  PF   32  CHI  23   0  25.7  5.1   8.7  0.587  0.2   
494         Trae Young  PG   22  ATL  26  26  34.7  7.6  17.7  0.429  2.3   
495        Cody Zeller   C   28  CHO  15  12  24.1  4.2   8.0  0.525  0.2   
496        Ivica Zubac   C   23  LAC  30   1  19.5  3.2   4.7  0.669  0.0   

     3PA    3P%   2P   2PA    2P%   eFG%   FT   FTA    FT%  ORB  DRB  TRB  

In [3010]:
np.shape(df1)

(497, 30)

Ya tenemos limpio el dataset de la temporada 2020-2021. Lo guardaremos para más adelante, cuando hagamos el merge de las temporadas.

In [3011]:
df1.to_csv("Temporada2020_2021.csv",sep=",")

# ANALISIS TEMPORADA 2021 - 2022

In [3012]:
# importación de los datos.
df2 = pd.read_csv("NBAPlayerStatsRegular.csv", encoding='latin-1', delimiter=';') 

In [3013]:
print(df2.head(11))

    Rk                    Player Pos  Age   Tm   G  GS    MP   FG   FGA  \
0    1          Precious Achiuwa   C   22  TOR  73  28  23.6  3.6   8.3   
1    2              Steven Adams   C   28  MEM  76  75  26.3  2.8   5.1   
2    3               Bam Adebayo   C   24  MIA  56  56  32.6  7.3  13.0   
3    4              Santi Aldama  PF   21  MEM  32   0  11.3  1.7   4.1   
4    5         LaMarcus Aldridge   C   36  BRK  47  12  22.3  5.4   9.7   
5    6  Nickeil Alexander-Walker  SG   23  TOT  65  21  22.6  3.9  10.5   
6    6  Nickeil Alexander-Walker  SG   23  NOP  50  19  26.3  4.7  12.6   
7    6  Nickeil Alexander-Walker  SG   23  UTA  15   2   9.9  1.1   3.2   
8    7             Grayson Allen  SG   26  MIL  66  61  27.3  3.9   8.6   
9    8             Jarrett Allen   C   23  CLE  56  56  32.3  6.6   9.7   
10   9             Jose Alvarado  PG   23  NOP  54   1  15.4  2.4   5.4   

      FG%   3P  3PA    3P%   2P   2PA    2P%   eFG%   FT  FTA    FT%  ORB  \
0   0.439  0.8  2.1  0

In [3014]:
# Información general sobre el dataframe
df2.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 812 entries, 0 to 811
Data columns (total 30 columns):
 #   Column  Non-Null Count  Dtype  
---  ------  --------------  -----  
 0   Rk      812 non-null    int64  
 1   Player  812 non-null    object 
 2   Pos     812 non-null    object 
 3   Age     812 non-null    int64  
 4   Tm      812 non-null    object 
 5   G       812 non-null    int64  
 6   GS      812 non-null    int64  
 7   MP      812 non-null    float64
 8   FG      812 non-null    float64
 9   FGA     812 non-null    float64
 10  FG%     812 non-null    float64
 11  3P      812 non-null    float64
 12  3PA     812 non-null    float64
 13  3P%     812 non-null    float64
 14  2P      812 non-null    float64
 15  2PA     812 non-null    float64
 16  2P%     812 non-null    float64
 17  eFG%    812 non-null    float64
 18  FT      812 non-null    float64
 19  FTA     812 non-null    float64
 20  FT%     812 non-null    float64
 21  ORB     812 non-null    float64
 22  DR

In [3015]:
np.shape(df2)

(812, 30)

In [3016]:
# consulta de valores nulos
ValoresNull2 = df2.isnull().sum() [df2.isnull().sum()>0]
print(ValoresNull2)

Series([], dtype: int64)


Revisando cantidad de veces que se repite un nombre.

In [3017]:
conteo_unicos = df2["Player"].value_counts()
print(conteo_unicos)

Player
Greg Monroe            5
Danuel House Jr.       4
Wenyen Gabriel         4
James Ennis III        4
Juancho Hernangómez    4
                      ..
Eric Bledsoe           1
Trae Young             1
Omer Yurtseven         1
Cody Zeller            1
Precious Achiuwa       1
Name: count, Length: 605, dtype: int64


In [3018]:
# Año que deseas agregar
Temporada2 = 2022

# Agregar una nueva columna 'Temporada' con el valor del año especificado
df2['Temp'] = Temporada2

# Mostrar las primeras filas del DataFrame para verificar
print(df2.head())

   Rk             Player Pos  Age   Tm   G  GS    MP   FG   FGA    FG%   3P  \
0   1   Precious Achiuwa   C   22  TOR  73  28  23.6  3.6   8.3  0.439  0.8   
1   2       Steven Adams   C   28  MEM  76  75  26.3  2.8   5.1  0.547  0.0   
2   3        Bam Adebayo   C   24  MIA  56  56  32.6  7.3  13.0  0.557  0.0   
3   4       Santi Aldama  PF   21  MEM  32   0  11.3  1.7   4.1  0.402  0.2   
4   5  LaMarcus Aldridge   C   36  BRK  47  12  22.3  5.4   9.7  0.550  0.3   

   3PA    3P%   2P   2PA    2P%   eFG%   FT  FTA    FT%  ORB  DRB   TRB  AST  \
0  2.1  0.359  2.9   6.1  0.468  0.486  1.1  1.8  0.595  2.0  4.5   6.5  1.1   
1  0.0  0.000  2.8   5.0  0.548  0.547  1.4  2.6  0.543  4.6  5.4  10.0  3.4   
2  0.1  0.000  7.3  12.9  0.562  0.557  4.6  6.1  0.753  2.4  7.6  10.1  3.4   
3  1.5  0.125  1.5   2.6  0.560  0.424  0.6  1.0  0.625  1.0  1.7   2.7  0.7   
4  1.0  0.304  5.1   8.8  0.578  0.566  1.9  2.2  0.873  1.6  3.9   5.5  0.9   

   STL  BLK  TOV   PF   PTS  Temp  
0  0.5  

In [3019]:
df_filtered = df2

Ahora notamos que ya quedo resuelto el tema de la columna TOTAL TEAM

In [3020]:
df_filtered.head(20)

Rk                    Player Pos  Age   Tm   G  GS    MP    FG   FGA  \
0    1          Precious Achiuwa   C   22  TOR  73  28  23.6   3.6   8.3   
1    2              Steven Adams   C   28  MEM  76  75  26.3   2.8   5.1   
2    3               Bam Adebayo   C   24  MIA  56  56  32.6   7.3  13.0   
3    4              Santi Aldama  PF   21  MEM  32   0  11.3   1.7   4.1   
4    5         LaMarcus Aldridge   C   36  BRK  47  12  22.3   5.4   9.7   
5    6  Nickeil Alexander-Walker  SG   23  TOT  65  21  22.6   3.9  10.5   
6    6  Nickeil Alexander-Walker  SG   23  NOP  50  19  26.3   4.7  12.6   
7    6  Nickeil Alexander-Walker  SG   23  UTA  15   2   9.9   1.1   3.2   
8    7             Grayson Allen  SG   26  MIL  66  61  27.3   3.9   8.6   
9    8             Jarrett Allen   C   23  CLE  56  56  32.3   6.6   9.7   
10   9             Jose Alvarado  PG   23  NOP  54   1  15.4   2.4   5.4   
11  10           Justin Anderson  SF   28  TOT  16   6  19.8   2.3   5.9   
12  10           Justin Anderson  SF   28  CLE   3   0  15.7   1.3   2.7   
13  10           Justin Anderson  SF   28  IND  13   6  20.7   2.5   6.7   
14  11             Kyle Anderson  PF   28  MEM  69  11  21.5   3.0   6.8   
15  12     Giannis Antetokounmpo  PF   27  MIL  67  67  32.9  10.3  18.6   
16  13    Thanasis Antetokounmpo  SF   29  MIL  48   6   9.9   1.5   2.7   
17  14           Carmelo Anthony  PF   37  LAL  69   3  26.0   4.6  10.5   
18  15              Cole Anthony  PG   21  ORL  65  65  31.7   5.5  14.0   
19  16                OG Anunoby  SF   24  TOR  48  48  36.0   6.4  14.5   

      FG%   3P  3PA    3P%   2P   2PA    2P%   eFG%   FT   FTA    FT%  ORB  \
0   0.439  0.8  2.1  0.359  2.9   6.1  0.468  0.486  1.1   1.8  0.595  2.0   
1   0.547  0.0  0.0  0.000  2.8   5.0  0.548  0.547  1.4   2.6  0.543  4.6   
2   0.557  0.0  0.1  0.000  7.3  12.9  0.562  0.557  4.6   6.1  0.753  2.4   
3   0.402  0.2  1.5  0.125  1.5   2.6  0.560  0.424  0.6   1.0  0.625  1.0   
4   0.550  0.3  1.0  0.304  5.1   8.8  0.578  0.566  1.9   2.2  0.873  1.6   
5   0.372  1.6  5.2  0.311  2.3   5.3  0.433  0.449  1.2   1.7  0.743  0.6   
6   0.375  1.9  6.1  0.311  2.8   6.5  0.434  0.450  1.4   1.9  0.722  0.7   
7   0.333  0.7  2.2  0.303  0.4   1.0  0.400  0.438  0.7   0.8  0.917  0.1   
8   0.448  2.4  5.9  0.409  1.5   2.7  0.533  0.588  1.0   1.1  0.865  0.5   
9   0.677  0.0  0.2  0.100  6.6   9.6  0.688  0.678  2.9   4.2  0.708  3.4   
10  0.446  0.6  2.0  0.291  1.8   3.4  0.538  0.500  0.7   1.0  0.679  0.5   
11  0.379  0.9  3.7  0.254  1.3   2.3  0.583  0.458  0.9   1.2  0.789  0.3   
12  0.500  0.7  2.0  0.333  0.7   0.7  1.000  0.625  1.0   1.3  0.750  0.0   
13  0.368  1.0  4.1  0.245  1.5   2.6  0.559  0.443  0.9   1.2  0.800  0.3   
14  0.446  0.5  1.6  0.330  2.5   5.2  0.481  0.484  1.0   1.5  0.638  1.0   
15  0.553  1.1  3.6  0.293  9.2  15.0  0.616  0.582  8.3  11.4  0.722  2.0   
16  0.547  0.0  0.3  0.143  1.4   2.4  0.596  0.555  0.6   1.0  0.630  0.8   
17  0.441  2.2  5.8  0.375  2.5   4.7  0.521  0.544  1.9   2.3  0.830  0.9   
18  0.391  2.0  6.0  0.338  3.5   8.0  0.432  0.464  3.3   3.9  0.854  0.5   
19  0.443  2.4  6.6  0.363  4.0   7.9  0.511  0.526  1.9   2.5  0.754  1.5   

    DRB   TRB  AST  STL  BLK  TOV   PF   PTS  Temp  
0   4.5   6.5  1.1  0.5  0.6  1.2  2.1   9.1  2022  
1   5.4  10.0  3.4  0.9  0.8  1.5  2.0   6.9  2022  
2   7.6  10.1  3.4  1.4  0.8  2.6  3.1  19.1  2022  
3   1.7   2.7  0.7  0.2  0.3  0.5  1.1   4.1  2022  
4   3.9   5.5  0.9  0.3  1.0  0.9  1.7  12.9  2022  
5   2.3   2.9  2.4  0.7  0.4  1.4  1.6  10.6  2022  
6   2.6   3.3  2.8  0.8  0.4  1.7  1.8  12.8  2022  
7   1.5   1.5  1.1  0.3  0.3  0.5  1.0   3.5  2022  
8   2.9   3.4  1.5  0.7  0.3  0.7  1.5  11.1  2022  
9   7.3  10.8  1.6  0.8  1.3  1.7  1.7  16.1  2022  
10  1.4   1.9  2.8  1.3  0.1  0.7  1.4   6.1  2022  
11  2.6   2.9  2.1  0.5  0.4  0.5  1.4   6.4  2022  
12  2.0   2.0  2.0  0.3  0.0  0.7  1.3   4.3  2022  
13  2.8   3.1  2.1  0.5  0

Consultamos si ahora los nombres se repiten

In [3021]:
conteo_unicos = df_filtered["Player"].value_counts()
print(conteo_unicos)

Player
Greg Monroe            5
Danuel House Jr.       4
Wenyen Gabriel         4
James Ennis III        4
Juancho Hernangómez    4
                      ..
Eric Bledsoe           1
Trae Young             1
Omer Yurtseven         1
Cody Zeller            1
Precious Achiuwa       1
Name: count, Length: 605, dtype: int64


In [3022]:
np.shape(df_filtered)

(812, 31)

Dropeamos la columna de los indices, porque son incorrectos.

In [3023]:
df_filtered = df_filtered.drop(columns=['Rk'])

In [3024]:
nuevos_nombres = {
    'Tm': 'Team',
    'G': 'GP'
}

# Renombrar las columnas especificadas
df_filtered.rename(columns=nuevos_nombres, inplace=True)


# Mostrar las primeras filas del DataFrame para verificar los cambios
print(df_filtered.head())

              Player Pos  Age Team  GP  GS    MP   FG   FGA    FG%   3P  3PA  \
0   Precious Achiuwa   C   22  TOR  73  28  23.6  3.6   8.3  0.439  0.8  2.1   
1       Steven Adams   C   28  MEM  76  75  26.3  2.8   5.1  0.547  0.0  0.0   
2        Bam Adebayo   C   24  MIA  56  56  32.6  7.3  13.0  0.557  0.0  0.1   
3       Santi Aldama  PF   21  MEM  32   0  11.3  1.7   4.1  0.402  0.2  1.5   
4  LaMarcus Aldridge   C   36  BRK  47  12  22.3  5.4   9.7  0.550  0.3  1.0   

     3P%   2P   2PA    2P%   eFG%   FT  FTA    FT%  ORB  DRB   TRB  AST  STL  \
0  0.359  2.9   6.1  0.468  0.486  1.1  1.8  0.595  2.0  4.5   6.5  1.1  0.5   
1  0.000  2.8   5.0  0.548  0.547  1.4  2.6  0.543  4.6  5.4  10.0  3.4  0.9   
2  0.000  7.3  12.9  0.562  0.557  4.6  6.1  0.753  2.4  7.6  10.1  3.4  1.4   
3  0.125  1.5   2.6  0.560  0.424  0.6  1.0  0.625  1.0  1.7   2.7  0.7  0.2   
4  0.304  5.1   8.8  0.578  0.566  1.9  2.2  0.873  1.6  3.9   5.5  0.9  0.3   

   BLK  TOV   PF   PTS  Temp  
0  0.6 

In [3025]:
# Obtener los valores únicos de la columna team_abbreviation
valores_unicos = df_filtered['Team'].unique()

# Mostrar los valores únicos
print(valores_unicos)

['TOR' 'MEM' 'MIA' 'BRK' 'TOT' 'NOP' 'UTA' 'MIL' 'CLE' 'IND' 'LAL' 'ORL'
 'NYK' 'HOU' 'WAS' 'PHO' 'SAC' 'DET' 'CHO' 'CHI' 'ATL' 'DEN' 'PHI' 'SAS'
 'LAC' 'OKC' 'MIN' 'DAL' 'GSW' 'POR' 'BOS']


In [3026]:
# Filtrar filas donde 'Team' es 'TOT'
df_tot = df_filtered[df_filtered['Team'] == 'TOT']

# Mostrar el resultado
print(df_tot)

                       Player Pos  Age Team  GP  GS    MP   FG   FGA    FG%  \
5    Nickeil Alexander-Walker  SG   23  TOT  65  21  22.6  3.9  10.5  0.372   
11            Justin Anderson  SF   28  TOT  16   6  19.8  2.3   5.9  0.379   
22              D.J. Augustin  PG   34  TOT  55   2  16.1  1.7   4.0  0.423   
29          Marvin Bagley III  PF   22  TOT  48  25  23.9  4.7   9.3  0.504   
51            DeAndre' Bembry  SF   27  TOT  56  20  18.3  2.2   3.9  0.561   
..                        ...  ..  ...  ...  ..  ..   ...  ...   ...    ...   
768            Tremont Waters  PG   24  TOT   3   0  16.7  1.3   4.7  0.286   
776             Derrick White  SG   27  TOT  75  52  29.3  4.5  10.7  0.421   
793           Justise Winslow  SF   25  TOT  48  11  16.1  2.3   5.4  0.428   
801              Moses Wright  PF   23  TOT   4   0   3.5  0.3   1.0  0.250   
805            Thaddeus Young  PF   33  TOT  52   1  16.3  2.7   5.2  0.518   

      3P  3PA    3P%   2P  2PA    2P%   eFG%   FT  

In [3027]:
# Obtener los valores únicos de la columna team_abbreviation
valores_unicos = df_tot['Player'].unique()

# Mostrar los valores únicos
print(valores_unicos)

['Nickeil Alexander-Walker' 'Justin Anderson' 'D.J. Augustin'
 'Marvin Bagley III' "DeAndre' Bembry" 'D?vis Bert?ns' 'Armoni Brooks'
 'Charlie Brown Jr.' 'Chaundee Brown Jr.' 'Moses Brown' 'Vernon Carey Jr.'
 'Jevon Carter' 'Willie Cauley-Stein' 'DeMarcus Cousins'
 'Robert Covington' 'Torrey Craig' 'Seth Curry' 'Spencer Dinwiddie'
 'Donte DiVincenzo' 'Jeff Dowtin' 'Goran Dragi?' 'Andre Drummond'
 'James Ennis III' 'Drew Eubanks' 'Bruno Fernando' 'Malik Fitts'
 'Bryn Forbes' 'Tim Frazier' 'Wenyen Gabriel' 'Langston Galloway'
 'Tyrese Haliburton' 'James Harden' 'Montrezl Harrell' 'Josh Hart'
 'Juancho Hernangómez' 'Buddy Hield' 'Malcolm Hill' 'Aaron Holiday'
 'Justin Holiday' 'Rodney Hood' 'Danuel House Jr.' 'Elijah Hughes'
 'Serge Ibaka' 'Josh Jackson' 'Justin Jackson' 'Alize Johnson'
 'Keon Johnson' 'Tyler Johnson' 'Carlik Jones' 'DeAndre Jordan'
 'Georgios Kalaitzakis' 'Braxton Key' 'Kevin Knox' 'Luke Kornet'
 'Jeremy Lamb' 'Romeo Langford' 'Caris LeVert' 'Didi Louzada' 'Trey Lyles'
 

In [3028]:
# Mapeo de jugadores a equipos
player_to_team = {
    'Nickeil Alexander-Walker': 'NOP',
    'Justin Anderson': 'CLE',
    'D.J. Augustin': 'HOU',
    'Marvin Bagley III': 'SAC',
    'DeAndre\' Bembry': 'BKN',
    'Davis Bertans': 'WSH',
    'Armoni Brooks': 'HOU',
    'Charlie Brown Jr.': 'ATL',
    'Chaundee Brown Jr.': 'DET',
    'Moses Brown': 'CLE',
    'Vernon Carey Jr.': 'WAS',
    'Jevon Carter': 'PHX',
    'Willie Cauley-Stein': 'DAL',
    'DeMarcus Cousins': 'DEN',
    'Robert Covington': 'POR',
    'Torrey Craig': 'PHX',
    'Seth Curry': 'DAL',
    'Spencer Dinwiddie': 'WAS',
    'Donte DiVincenzo': 'MIL',
    'Jeff Dowtin': 'TOR',
    'Goran Dragić': 'TOR',
    'Andre Drummond': 'PHI',
    'James Ennis III': 'ORL',
    'Drew Eubanks': 'POR',
    'Bruno Fernando': 'BOS',
    'Malik Fitts': 'UTA',
    'Bryn Forbes': 'MIL',
    'Tim Frazier': 'NOP',
    'Wenyen Gabriel': 'LAL',
    'Langston Galloway': 'PHX',
    'Tyrese Haliburton': 'SAC',
    'James Harden': 'BKN',
    'Montrezl Harrell': 'WAS',
    'Josh Hart': 'NOP',
    'Juancho Hernangómez': 'BOS',
    'Buddy Hield': 'SAC',
    'Malcolm Hill': 'ATL',
    'Aaron Holiday': 'IND',
    'Justin Holiday': 'IND',
    'Rodney Hood': 'POR',
    'Danuel House Jr.': 'HOU',
    'Elijah Hughes': 'UTA',
    'Serge Ibaka': 'LAC',
    'Josh Jackson': 'DET',
    'Justin Jackson': 'OKC',
    'Alize Johnson': 'BKN',
    'Keon Johnson': 'LAC',
    'Tyler Johnson': 'BKN',
    'Carlik Jones': 'DAL',
    'DeAndre Jordan': 'LAL',
    'Georgios Kalaitzakis': 'MIL',
    'Braxton Key': 'DET',
    'Kevin Knox': 'NYK',
    'Luke Kornet': 'BOS',
    'Jeremy Lamb': 'IND',
    'Romeo Langford': 'BOS',
    'Caris LeVert': 'IND',
    'Didi Louzada': 'NOP',
    'Trey Lyles': 'SAC',
    'Kelan Martin': 'IND',
    'Mac McClung': 'GSW',
    'CJ McCollum': 'POR',
    'Paul Millsap': 'BKN',
    'Greg Monroe': 'TOR',
    'Juwan Morgan': 'BOS',
    'Mychal Mulder': 'GSW',
    'Larry Nance Jr.': 'POR',
    'Semi Ojeleye': 'BOS',
    'Reggie Perry': 'BKN',
    'Kristaps Porzingis': 'DAL',
    'Norman Powell': 'TOR',
    'Cam Reddish': 'ATL',
    'Josh Richardson': 'BOS',
    'Justin Robinson': 'WAS',
    'Rajon Rondo': 'LAL',
    'Domantas Sabonis': 'IND',
    'Tomáš Satoranský': 'NOP',
    'Dennis Schröder': 'BOS',
    'Chris Silva': 'MIA',
    'Javonte Smart': 'BOS',
    'Ish Smith': 'WAS',
    'Jalen Smith': 'PHX',
    'Xavier Sneed': 'GSW',
    'Tony Snell': 'ATL',
    'Nik Stauskas': 'CLE',
    'Lance Stephenson': 'IND',
    'Daniel Theis': 'BOS',
    'Isaiah Thomas': 'LAL',
    'Tristan Thompson': 'BOS',
    'Rayjon Tucker': 'UTA',
    'Denzel Valentine': 'CHI',
    'Brad Wanamaker': 'GSW',
    'Tremont Waters': 'BOS',
    'Derrick White': 'SAS',
    'Justise Winslow': 'MEM',
    'Moses Wright': 'DAL',
    'Thaddeus Young': 'SAS'
}

# Reemplazar 'TOT' por el equipo correspondiente
def replace_team(row):
    if 'TOT' in row['Team']:
        return player_to_team.get(row['Player'], row['Team'])  # Reemplaza si hay un mapeo, de lo contrario mantiene el valor original
    return row['Team']

df_filtered['Team'] = df_filtered.apply(replace_team, axis=1)

print(df_filtered)

                Player Pos  Age Team  GP  GS    MP   FG   FGA    FG%   3P  \
0     Precious Achiuwa   C   22  TOR  73  28  23.6  3.6   8.3  0.439  0.8   
1         Steven Adams   C   28  MEM  76  75  26.3  2.8   5.1  0.547  0.0   
2          Bam Adebayo   C   24  MIA  56  56  32.6  7.3  13.0  0.557  0.0   
3         Santi Aldama  PF   21  MEM  32   0  11.3  1.7   4.1  0.402  0.2   
4    LaMarcus Aldridge   C   36  BRK  47  12  22.3  5.4   9.7  0.550  0.3   
..                 ...  ..  ...  ...  ..  ..   ...  ...   ...    ...  ...   
807     Thaddeus Young  PF   33  TOR  26   0  18.3  2.6   5.5  0.465  0.7   
808         Trae Young  PG   23  ATL  76  76  34.9  9.4  20.3  0.460  3.1   
809     Omer Yurtseven   C   23  MIA  56  12  12.6  2.3   4.4  0.526  0.0   
810        Cody Zeller   C   29  POR  27   0  13.1  1.9   3.3  0.567  0.0   
811        Ivica Zubac   C   24  LAC  76  76  24.4  4.1   6.5  0.626  0.0   

     3PA    3P%   2P   2PA    2P%   eFG%   FT  FTA    FT%  ORB  DRB   TRB  

In [3029]:
# Filtrar filas donde 'Team' es 'TOT'
df_tot = df_filtered[df_filtered['Team'] == 'TOT']

# Mostrar el resultado
print(df_tot)

                 Player Pos  Age Team  GP  GS    MP   FG   FGA    FG%   3P  \
54        D?vis Bert?ns  PF   29  TOT  56   0  14.4  1.8   4.9  0.360  1.4   
189        Goran Dragi?  PG   35  TOT  21   8  23.7  2.7   7.2  0.377  0.8   
601  Kristaps Porzi??is  PF   26  TOT  51  51  29.0  6.9  15.0  0.459  1.5   
651    TomᚠSatoransk?  SG   30  TOT  55  13  16.5  1.3   3.5  0.377  0.2   

     3PA    3P%   2P   2PA    2P%   eFG%   FT  FTA    FT%  ORB  DRB  TRB  AST  \
54   4.1  0.335  0.4   0.8  0.489  0.500  0.6  0.7  0.900  0.2  1.9  2.1  0.6   
189  3.0  0.254  2.0   4.2  0.466  0.430  1.3  1.6  0.818  0.6  2.5  3.1  4.1   
601  4.9  0.310  5.4  10.1  0.532  0.510  4.9  5.6  0.867  2.0  6.1  8.1  2.3   
651  1.0  0.208  1.1   2.5  0.442  0.406  0.8  1.0  0.796  0.5  1.8  2.3  3.3   

     STL  BLK  TOV   PF   PTS  Temp  
54   0.3  0.2  0.3  1.5   5.6  2022  
189  0.9  0.2  1.6  2.1   7.5  2022  
601  0.7  1.6  1.6  2.7  20.2  2022  
651  0.5  0.1  0.8  1.3   3.6  2022  


Guardamos la temporada ya con su respectivo proceso ETL.

In [3030]:
df_filtered.to_csv("Temporada2021_2022.csv",sep=",")

# ANALISIS TEMPORADA 2022 - 2023

In [3031]:
df3 = pd.read_csv("nba_salaries.csv") 
print(df3.head(11))

    Unnamed: 0            Player Name    Salary Position  Age     Team  GP  \
0            0          Stephen Curry  48070014       PG   34      GSW  56   
1            1              John Wall  47345760       PG   32      LAC  34   
2            2      Russell Westbrook  47080179       PG   34  LAL/LAC  73   
3            3           LeBron James  44474988       PF   38      LAL  55   
4            4           Kevin Durant  44119845       PF   34  BRK/PHO  47   
5            5           Bradley Beal  43279250       SG   29      WAS  50   
6            6          Kawhi Leonard  42492492       SF   31      LAC  52   
7            7            Paul George  42492492       SF   32      LAC  56   
8            8  Giannis Antetokounmpo  42492492       PF   28      MIL  63   
9            9         Damian Lillard  42492492       PG   32      POR  58   
10          10          Klay Thompson  40600080       SF   32      GSW  69   

    GS    MP    FG   FGA    FG%   3P   3PA    3P%    2P   2PA  

In [3032]:
df3.drop(df3.columns[0], axis=1, inplace=True)

In [3033]:
print(df3.head(11))

              Player Name    Salary Position  Age     Team  GP  GS    MP  \
0           Stephen Curry  48070014       PG   34      GSW  56  56  34.7   
1               John Wall  47345760       PG   32      LAC  34   3  22.2   
2       Russell Westbrook  47080179       PG   34  LAL/LAC  73  24  29.1   
3            LeBron James  44474988       PF   38      LAL  55  54  35.5   
4            Kevin Durant  44119845       PF   34  BRK/PHO  47  47  35.6   
5            Bradley Beal  43279250       SG   29      WAS  50  50  33.5   
6           Kawhi Leonard  42492492       SF   31      LAC  52  50  33.6   
7             Paul George  42492492       SF   32      LAC  56  56  34.6   
8   Giannis Antetokounmpo  42492492       PF   28      MIL  63  63  32.1   
9          Damian Lillard  42492492       PG   32      POR  58  58  36.3   
10          Klay Thompson  40600080       SF   32      GSW  69  69  33.0   

      FG   FGA    FG%   3P   3PA    3P%    2P   2PA    2P%   eFG%   FT   FTA  \
0   10.

In [3034]:
print(df3.describe())

             Salary         Age          GP          GS          MP  \
count  4.670000e+02  467.000000  467.000000  467.000000  467.000000   
mean   8.416599e+06   25.820128   48.233405   22.650964   19.871306   
std    1.070812e+07    4.275113   24.807740   27.094577    9.548684   
min    5.849000e+03   19.000000    1.000000    0.000000    1.800000   
25%    1.782621e+06   23.000000   31.000000    1.000000   12.500000   
50%    3.722040e+06   25.000000   55.000000    8.000000   19.200000   
75%    1.063354e+07   29.000000   68.500000   45.500000   28.300000   
max    4.807001e+07   42.000000   83.000000   83.000000   41.000000   

               FG         FGA         FG%          3P         3PA         3P%  \
count  467.000000  467.000000  466.000000  467.000000  467.000000  454.000000   
mean     3.351392    7.117773    0.465009    0.996574    2.792719    0.325090   
std      2.457836    5.020700    0.109547    0.880468    2.260794    0.131624   
min      0.000000    0.000000    0.0

Se eliminan las columnas que no sirven. En este caso contenian información no relevante.

In [3035]:
df3.drop(df3.columns[30], axis=1, inplace=True)

In [3036]:
df3.drop(df3.columns[1], axis=1, inplace=True)

In [3037]:
df3.head(11)

Player Name Position  Age     Team  GP  GS    MP    FG   FGA  \
0           Stephen Curry       PG   34      GSW  56  56  34.7  10.0  20.2   
1               John Wall       PG   32      LAC  34   3  22.2   4.1   9.9   
2       Russell Westbrook       PG   34  LAL/LAC  73  24  29.1   5.9  13.6   
3            LeBron James       PF   38      LAL  55  54  35.5  11.1  22.2   
4            Kevin Durant       PF   34  BRK/PHO  47  47  35.6  10.3  18.3   
5            Bradley Beal       SG   29      WAS  50  50  33.5   8.9  17.6   
6           Kawhi Leonard       SF   31      LAC  52  50  33.6   8.6  16.8   
7             Paul George       SF   32      LAC  56  56  34.6   8.2  17.9   
8   Giannis Antetokounmpo       PF   28      MIL  63  63  32.1  11.2  20.3   
9          Damian Lillard       PG   32      POR  58  58  36.3   9.6  20.7   
10          Klay Thompson       SF   32      GSW  69  69  33.0   7.9  18.1   

      FG%   3P   3PA    3P%    2P   2PA    2P%   eFG%   FT   FTA    FT%  ORB  \
0   0.493  4.9  11.4  0.427   5.1   8.8  0.579  0.614  4.6   5.0  0.915  0.7   
1   0.408  1.0   3.2  0.303   3.1   6.7  0.459  0.457  2.3   3.3  0.681  0.4   
2   0.436  1.2   3.9  0.311   4.7   9.7  0.487  0.481  2.8   4.3  0.656  1.2   
3   0.500  2.2   6.9  0.321   8.9  15.3  0.580  0.549  4.6   5.9  0.768  1.2   
4   0.560  2.0   4.9  0.404   8.3  13.4  0.617  0.614  6.5   7.1  0.919  0.4   
5   0.506  1.6   4.4  0.365   7.3  13.2  0.552  0.551  3.8   4.6  0.842  0.8   
6   0.512  2.0   4.8  0.416   6.6  11.9  0.551  0.572  4.7   5.4  0.871  1.1   
7   0.457  2.8   7.6  0.371   5.4  10.3  0.521  0.536  4.6   5.3  0.871  0.8   
8   0.553  0.7   2.7  0.275  10.5  17.6  0.596  0.572  7.9  12.3  0.645  2.2   
9   0.463  4.2  11.3  0.371   5.4   9.4  0.574  0.564  8.8   9.6  0.914  0.8   
10  0.436  4.4  10.6  0.412   3.6   7.6  0.470  0.556  1.7   1.9  0.879  0.6   

    DRB   TRB  AST  STL  BLK  TOV   PF   PTS  
0   5.4   6.1  6.3  0.9  0.4  3.2  2.1  29.4  
1   2.3   2.7  5.2  0.8  0.4  2.4  1.7  11.4  
2   4.6   5.8  7.5  1.0  0.5  3.5  2.2  15.9  
3   7.1   8.3  6.8  0.9  0.6  3.2  1.6  28.9  
4   6.3   6.7  5.0  0.7  1.4  3.3  2.1  29.1  
5   3.1   3.9  5.4  0.9  0.7  2.9  2.1  23.2  
6   5.4   6.5  3.9  1.4  0.5  1.7  1.6  23.8  
7   5.3   6.1  5.1  1.5  0.4  3.1  2.8  23.8  
8   9.6  11.8  5.7  0.8  0.8  3.9  3.1  31.1  
9   4.0   4.8  7.3  0.9  0.3  3.3  1.9  32.2  
10  3.6   4.1  2.4  0.7  0.4  1.8  1.9  21.9

Verificamos valores nulos

In [3038]:
ValoresNull3 = df3.isnull().sum() [df3.isnull().sum()>0]
print(ValoresNull3)

FG%      1
3P%     13
2P%      4
eFG%     1
FT%     23
dtype: int64


In [3039]:
# Filtrar filas que contienen al menos un valor nulo
filas_con_nulos = df3[df3.isnull().any(axis=1)]

# Mostrar las primeras filas con valores nulos

np.shape(filas_con_nulos)

(34, 29)

Observamos los valores nulos. Es la primer temporada en presentar valores nulos. A continuación en base a la naturaleza de los nulos fijaremos parámetros para tratarlos.

In [3040]:
filas_con_nulos

Player Name Position  Age     Team  GP  GS    MP   FG  FGA    FG%  \
75   Mitchell Robinson        C   24      NYK  59  58  27.0  3.2  4.7  0.671   
133      Mason Plumlee        C   32  LAC/CHO  79  60  26.0  4.2  6.1  0.680   
221      Dylan Windler       SF   26      CLE   3   0   3.3  0.7  1.0  0.667   
233      Mark Williams        C   21      CHO  43  17  19.3  3.7  5.8  0.637   
247     Sterling Brown       SF   27      LAL   4   0   6.0  0.0  1.0  0.000   
254       Joe Wieskamp       SF   23      TOR   9   0   5.6  0.3  1.6  0.214   
264    Bismack Biyombo        C   30      PHO  61  14  14.3  2.0  3.4  0.578   
301     Udoka Azubuike        C   23      UTA  36   4  10.0  1.6  2.0  0.819   
332     Daniel Gafford        C   24      WAS  78  47  20.6  3.7  5.1  0.732   
344   Ryan Arcidiacono       PG   28  NYK/POR  20   4   8.6  0.5  1.9  0.243   
356    Leandro Bolmaro       SF   22      UTA  14   0   4.9  0.2  1.4  0.150   
358       Jared Butler       PG   22      OKC   6   1  12.8  2.5  5.3  0.469   
393  Justin Champagnie       SF   21  BOS/TOR   5   0   6.8  1.0  3.0  0.333   
395        Moses Brown        C   23  BRK/LAC  36   1   8.2  1.7  2.7  0.635   
397          PJ Dozier       SF   26      SAC  16   0   4.9  0.6  2.1  0.303   
403       Trevor Keels       SG   19      NYK   3   0   2.7  0.3  1.3  0.250   
404     Dereon Seabron       SG   22      NOP   5   0   2.4  0.4  1.0  0.400   
408       Malcolm Hill       SF   27      CHI   5   0   1.8  0.4  0.8  0.500   
416        Cole Swider       SF   23      LAL   7   0   5.9  0.4  1.3  0.333   
421      Neemias Queta        C   23      SAC   5   0   5.8  1.2  1.8  0.667   
425       Micah Potter       PF   24      UTA   7   0   7.4  1.4  2.1  0.667   
428       Chima Moneke       SF   27      SAC   2   0   4.0  0.5  1.0  0.500   
436   Alondes Williams       SG   23      BRK   1   0   5.0  0.0  0.0    NaN   
437   Donovan Williams       SG   21      ATL   2   0   2.0  1.0  2.5  0.400   
439        Chris Silva       PF   26      DAL   1   0   3.0  1.0  1.0  1.000   
440       Tyler Dorsey       SF   26      DAL   3   0   2.7  1.3  1.7  0.800   
441        Braxton Key       SF   25      DET   3   0   3.0  0.3  0.3  1.000   
444       Devon Dotson       PG   23      WAS   6   0   8.8  0.2  1.7  0.100   
449        Xavier Moon       SG   28      LAC   4   0   5.0  0.8  2.3  0.333   
453      Frank Jackson       PG   24      UTA   1   0   5.0  0.0  3.0  0.000   
454      Deonte Burton       PF   29      SAC   2   0   3.0  0.0  1.0  0.000   
458     Jordan Schakel       SF   24      WAS   2   0   3.0  0.5  1.0  0.500   
459      Stanley Umude       SG   23      DET   1   0   2.0  0.0  1.0  0.000   
466      Jacob Gilyard       PG   24      MEM   1   0  41.0  1.0  3.0  0.333   

      3P  3PA    3P%   2P  2PA    2P%   eFG%   FT  FTA    FT%  ORB  DRB  TRB  \
75   0.0  0.0    NaN  3.2  4.7  0.671  0.671  1.0  2.1  0.484  4.5  4.9  9.4   
133  0.0  0.0    NaN  4.2  6.1  0.680  0.680  2.5  4.0  0.636  2.9  6.0  8.9   
221  0.3  0.7  0.500  0.3  0.3  1.000  0.833  0.0  0.0    NaN  0.0  0.0  0.0   
233  0.0  0.0    NaN  3.7  5.8  0.637  0.637  1.6  2.3  0.691  2.2  4.9  7.1   
247  0.0  0.5  0.000  0.0  0.5  0.000  0.000  0.0  0.0    NaN  0.8  1.3  2.0   
254  0.3  1.3  0.250  0.0  0.2  0.000  0.321  0.0  0.0    NaN  0.0  0.4  0.4   
264  0.0  0.0    NaN  2.0  3.4  0.578  0.578  0.4  1.1  0.357  1.5  2.8  4.3   
301  0.0  0.0    NaN  1.6  2.0  0.819  0.819  0.2  0.6  0.350  0.9  2.4  3.3   
332  0.0  0.0    NaN  3.7  5.1  0.732  0.732  1.6  2.4  0.679  2.1  3.5  5.6   
344  0.4  1.2  0.348  0.1  0.7  0.071  0.351  0.0  0.0    NaN  0.0  0.8  0.8   
356  0.0  0.3  0.000  0.2  1.1  0.188  0.150  0.0  0.0    NaN  0.3  0.2  0.5   
358  1.2  2.3  0.500  1.3  3.0  0.444  0.578  0.0  0.0    NaN  0.2  0.5  0.7   
393  0.2  1.0  0.200  0.8  2.0  0.400  0.367  0.0  0.0    NaN  0.6  1.0  1.6   
395  0.0  0.0    NaN  1.7  2.7  0.635  0.635  0.9  2.0  0.458  1.6  2.3  3.9   
397  0.1  

Consultaremos los datos oficiales de la página de la nba (www.nba.com)

Esto lo haremos columna por columna debido a la delicada naturaleza de cada estadistica.

In [3041]:
columnas_a_imprimir = ['Player Name','FG%']
print(filas_con_nulos[columnas_a_imprimir])

           Player Name    FG%
75   Mitchell Robinson  0.671
133      Mason Plumlee  0.680
221      Dylan Windler  0.667
233      Mark Williams  0.637
247     Sterling Brown  0.000
254       Joe Wieskamp  0.214
264    Bismack Biyombo  0.578
301     Udoka Azubuike  0.819
332     Daniel Gafford  0.732
344   Ryan Arcidiacono  0.243
356    Leandro Bolmaro  0.150
358       Jared Butler  0.469
393  Justin Champagnie  0.333
395        Moses Brown  0.635
397          PJ Dozier  0.303
403       Trevor Keels  0.250
404     Dereon Seabron  0.400
408       Malcolm Hill  0.500
416        Cole Swider  0.333
421      Neemias Queta  0.667
425       Micah Potter  0.667
428       Chima Moneke  0.500
436   Alondes Williams    NaN
437   Donovan Williams  0.400
439        Chris Silva  1.000
440       Tyler Dorsey  0.800
441        Braxton Key  1.000
444       Devon Dotson  0.100
449        Xavier Moon  0.333
453      Frank Jackson  0.000
454      Deonte Burton  0.000
458     Jordan Schakel  0.500
459      S

 Alondes Williams el unico valor faltante de aquí no tiene su estadística disponible en la página de la NBA entonces para no generar data errónea lo mejor será colocarle 0.

In [3042]:
# Rellenar valores faltantes de Alondes Williams con 0 en la fila con índice 436
alondes = [436]
df3.loc[alondes, 'FG%'] = df3.loc[alondes, 'FG%'].fillna(0)

In [3043]:
columnas_a_imprimir = ['Player Name','3P%']
print(filas_con_nulos[columnas_a_imprimir])

           Player Name    3P%
75   Mitchell Robinson    NaN
133      Mason Plumlee    NaN
221      Dylan Windler  0.500
233      Mark Williams    NaN
247     Sterling Brown  0.000
254       Joe Wieskamp  0.250
264    Bismack Biyombo    NaN
301     Udoka Azubuike    NaN
332     Daniel Gafford    NaN
344   Ryan Arcidiacono  0.348
356    Leandro Bolmaro  0.000
358       Jared Butler  0.500
393  Justin Champagnie  0.200
395        Moses Brown    NaN
397          PJ Dozier  0.125
403       Trevor Keels  0.250
404     Dereon Seabron    NaN
408       Malcolm Hill  0.333
416        Cole Swider  0.375
421      Neemias Queta    NaN
425       Micah Potter  0.571
428       Chima Moneke    NaN
436   Alondes Williams    NaN
437   Donovan Williams  0.000
439        Chris Silva    NaN
440       Tyler Dorsey  0.500
441        Braxton Key    NaN
444       Devon Dotson  0.250
449        Xavier Moon  0.333
453      Frank Jackson  0.000
454      Deonte Burton  0.000
458     Jordan Schakel  1.000
459      S

En el caso de estos valores faltantes, si figuran en la página oficial de NBA pero aparecen como 0%.

Esto refleja que hay jugadores que no tuvieron oportunidades de hacer tiros de 3 puntos.

 Procedo a rellenarlos con 0.

In [3044]:
indices = [75,133,233,264,301,332,395,404,421,428,436,441,439]
df3.loc[indices, '3P%'] = df3.loc[indices, '3P%'].fillna(0)

In [3045]:
columnas_a_imprimir = ['Player Name','2P%']
print(filas_con_nulos[columnas_a_imprimir])

           Player Name    2P%
75   Mitchell Robinson  0.671
133      Mason Plumlee  0.680
221      Dylan Windler  1.000
233      Mark Williams  0.637
247     Sterling Brown  0.000
254       Joe Wieskamp  0.000
264    Bismack Biyombo  0.578
301     Udoka Azubuike  0.819
332     Daniel Gafford  0.732
344   Ryan Arcidiacono  0.071
356    Leandro Bolmaro  0.188
358       Jared Butler  0.444
393  Justin Champagnie  0.400
395        Moses Brown  0.635
397          PJ Dozier  0.471
403       Trevor Keels    NaN
404     Dereon Seabron  0.400
408       Malcolm Hill  1.000
416        Cole Swider  0.000
421      Neemias Queta  0.667
425       Micah Potter  0.750
428       Chima Moneke  0.500
436   Alondes Williams    NaN
437   Donovan Williams  0.667
439        Chris Silva  1.000
440       Tyler Dorsey  1.000
441        Braxton Key  1.000
444       Devon Dotson  0.000
449        Xavier Moon  0.333
453      Frank Jackson  0.000
454      Deonte Burton  0.000
458     Jordan Schakel  0.000
459      S

Stanley Umude: No tuvo estadísticas de la NBA para la temporada 2022-2023, ya que no jugó en partidos de temporada regular.

Jacob Gilyard: Similar a Umude, Jacob Gilyard no jugó en la NBA durante la temporada 2022-2023, por lo que no tiene estadísticas de 2P%.

Trevor Keels: Trevor Keels, quien también fue novato en la temporada 2022-2023, jugó para los New York Knicks y tuvo un porcentaje de tiros de campo de dos puntos (2P%) de 33.3% en los partidos que disputó.

Alondes Williams: Alondes Williams tampoco tuvo estadísticas en la NBA durante la temporada 2022-2023, ya que no participó en partidos de temporada regular.

In [3046]:
indices = [466,459,436]
df3.loc[indices, '2P%'] = df3.loc[indices, '2P%'].fillna(0)
trevor_keels = [403]
df3.loc[trevor_keels, '2P%'] = df3.loc[trevor_keels, '2P%'].fillna(0.333)

In [3047]:
columnas_a_imprimir = ['Player Name','eFG%']
print(filas_con_nulos[columnas_a_imprimir])

           Player Name   eFG%
75   Mitchell Robinson  0.671
133      Mason Plumlee  0.680
221      Dylan Windler  0.833
233      Mark Williams  0.637
247     Sterling Brown  0.000
254       Joe Wieskamp  0.321
264    Bismack Biyombo  0.578
301     Udoka Azubuike  0.819
332     Daniel Gafford  0.732
344   Ryan Arcidiacono  0.351
356    Leandro Bolmaro  0.150
358       Jared Butler  0.578
393  Justin Champagnie  0.367
395        Moses Brown  0.635
397          PJ Dozier  0.333
403       Trevor Keels  0.375
404     Dereon Seabron  0.400
408       Malcolm Hill  0.625
416        Cole Swider  0.500
421      Neemias Queta  0.667
425       Micah Potter  0.800
428       Chima Moneke  0.500
436   Alondes Williams    NaN
437   Donovan Williams  0.400
439        Chris Silva  1.000
440       Tyler Dorsey  0.900
441        Braxton Key  1.000
444       Devon Dotson  0.150
449        Xavier Moon  0.389
453      Frank Jackson  0.000
454      Deonte Burton  0.000
458     Jordan Schakel  0.750
459      S

In [3048]:
alondes = [436]
df3.loc[alondes, 'eFG%'] = df3.loc[alondes, 'eFG%'].fillna(0)

In [3049]:
columnas_a_imprimir = ['Player Name','FT%']
print(filas_con_nulos[columnas_a_imprimir])

           Player Name    FT%
75   Mitchell Robinson  0.484
133      Mason Plumlee  0.636
221      Dylan Windler    NaN
233      Mark Williams  0.691
247     Sterling Brown    NaN
254       Joe Wieskamp    NaN
264    Bismack Biyombo  0.357
301     Udoka Azubuike  0.350
332     Daniel Gafford  0.679
344   Ryan Arcidiacono    NaN
356    Leandro Bolmaro    NaN
358       Jared Butler    NaN
393  Justin Champagnie    NaN
395        Moses Brown  0.458
397          PJ Dozier    NaN
403       Trevor Keels    NaN
404     Dereon Seabron    NaN
408       Malcolm Hill    NaN
416        Cole Swider    NaN
421      Neemias Queta  0.000
425       Micah Potter    NaN
428       Chima Moneke  0.000
436   Alondes Williams    NaN
437   Donovan Williams    NaN
439        Chris Silva    NaN
440       Tyler Dorsey    NaN
441        Braxton Key  1.000
444       Devon Dotson    NaN
449        Xavier Moon    NaN
453      Frank Jackson    NaN
454      Deonte Burton    NaN
458     Jordan Schakel    NaN
459      S

Estas fueron las stats oficiales (FT%) de la pagina NBA para cada jugador.

Dylan Windler: 70.4%

Sterling Brown: 78.6%

Joe Wieskamp: 71.4%

Ryan Arcidiacono: 80.0%

Leandro Bolmaro: 68.8%

Jared Butler: 76.2%

Justin Champagnie: 76.5%

P.J. Dozier: 69.4%

Trevor Keels: 63.6%

Dereon Seabron: 77.8%

Malcolm Hill: 81.8%

Cole Swider: — (No jugó en la NBA durante la temporada 2022-2023)

Micah Potter: — (No jugó en la NBA durante la temporada 2022-2023)

Alondes Williams: — (No jugó en la NBA durante la temporada 2022-2023)

Donovan Williams: — (No jugó en la NBA durante la temporada 2022-2023)

Chris Silva: 78.9%

Tyler Dorsey: 83.3%

Devon Dotson: 80.0%

Xavier Moon: 82.8%

Frank Jackson: 83.0%

Deonte Burton: 80.0%

Jordan Schakel: — (No jugó en la NBA durante la temporada 2022-2023)

Jacob Gilyard: — (No jugó en la NBA durante la temporada 2022-2023)


In [3050]:

# Diccionario con valores de reemplazo
reemplazos = {
    'Dylan Windler': 0.704,
    'Sterling Brown': 0.786,
    'Joe Wieskamp': 0.714,
    'Ryan Arcidiacono': 0.800,
    'Leandro Bolmaro': 0.688,
    'Jared Butler': 0.762,
    'Justin Champagnie': 0.765,
    'PJ Dozier': 0.694,
    'Trevor Keels': 0.636,
    'Dereon Seabron': 0.778,
    'Malcolm Hill': 0.818,
    'Cole Swider': 0,  # No jugó
    'Micah Potter': 0,  # No jugó
    'Alondes Williams': 0,  # No jugó
    'Donovan Williams': 0,  # No jugó
    'Chris Silva': 0.789,
    'Tyler Dorsey': 0.833,
    'Devon Dotson': 0.800,
    'Xavier Moon': 0.828,
    'Frank Jackson': 0.830,
    'Deonte Burton': 0.800,
    'Jordan Schakel': 0,  # No jugó
    'Jacob Gilyard': 0  # No jugó
}

# Rellenar los valores faltantes en la columna 'FT%' con el diccionario de reemplazo
df3['FT%'] = df3.apply(lambda row: reemplazos.get(row['Player Name'], row['FT%']), axis=1)

Checkeo nuevamente valores faltantes:

In [3051]:
ValoresNull3 = df3.isnull().sum() [df3.isnull().sum()>0]
print(ValoresNull3)

Series([], dtype: int64)


In [3052]:
# Lista de columnas para las que se reemplazarán los valores nulos con la media
columnas_objetivo = ['FG%', '3P%', '2P%', 'eFG%', 'FT%']

# Reemplazar los valores nulos de las columnas especificadas por su media
for columna in columnas_objetivo:
    media_columna = df3[columna].mean()
    df3[columna] = df3[columna].fillna(media_columna)

# Mostrar las primeras filas del DataFrame para verificar
print(df3.head(11))

              Player Name Position  Age     Team  GP  GS    MP    FG   FGA  \
0           Stephen Curry       PG   34      GSW  56  56  34.7  10.0  20.2   
1               John Wall       PG   32      LAC  34   3  22.2   4.1   9.9   
2       Russell Westbrook       PG   34  LAL/LAC  73  24  29.1   5.9  13.6   
3            LeBron James       PF   38      LAL  55  54  35.5  11.1  22.2   
4            Kevin Durant       PF   34  BRK/PHO  47  47  35.6  10.3  18.3   
5            Bradley Beal       SG   29      WAS  50  50  33.5   8.9  17.6   
6           Kawhi Leonard       SF   31      LAC  52  50  33.6   8.6  16.8   
7             Paul George       SF   32      LAC  56  56  34.6   8.2  17.9   
8   Giannis Antetokounmpo       PF   28      MIL  63  63  32.1  11.2  20.3   
9          Damian Lillard       PG   32      POR  58  58  36.3   9.6  20.7   
10          Klay Thompson       SF   32      GSW  69  69  33.0   7.9  18.1   

      FG%   3P   3PA    3P%    2P   2PA    2P%   eFG%   FT   FT

In [3053]:
# Mostrar el valor de la columna 'FT%' en la fila 221
valor_fila_221 = df3.loc[221, 'FT%']
print(f"Valor en la fila 221 de 'FT%': {valor_fila_221}")

# Mostrar el valor de la columna 'FT%' en la fila 247
valor_fila_247 = df3.loc[247, 'FT%']
print(f"Valor en la fila 247 de 'FT%': {valor_fila_247}")

Valor en la fila 221 de 'FT%': 0.704
Valor en la fila 247 de 'FT%': 0.786


In [3054]:
nuevos_nombres = {
    'Player Name': 'Player',
    'Position': 'Pos'
}

# Renombrar las columnas especificadas
df3.rename(columns=nuevos_nombres, inplace=True)


# Mostrar las primeras filas del DataFrame para verificar los cambios
print(df3.head())

              Player Pos  Age     Team  GP  GS    MP    FG   FGA    FG%   3P  \
0      Stephen Curry  PG   34      GSW  56  56  34.7  10.0  20.2  0.493  4.9   
1          John Wall  PG   32      LAC  34   3  22.2   4.1   9.9  0.408  1.0   
2  Russell Westbrook  PG   34  LAL/LAC  73  24  29.1   5.9  13.6  0.436  1.2   
3       LeBron James  PF   38      LAL  55  54  35.5  11.1  22.2  0.500  2.2   
4       Kevin Durant  PF   34  BRK/PHO  47  47  35.6  10.3  18.3  0.560  2.0   

    3PA    3P%   2P   2PA    2P%   eFG%   FT  FTA    FT%  ORB  DRB  TRB  AST  \
0  11.4  0.427  5.1   8.8  0.579  0.614  4.6  5.0  0.915  0.7  5.4  6.1  6.3   
1   3.2  0.303  3.1   6.7  0.459  0.457  2.3  3.3  0.681  0.4  2.3  2.7  5.2   
2   3.9  0.311  4.7   9.7  0.487  0.481  2.8  4.3  0.656  1.2  4.6  5.8  7.5   
3   6.9  0.321  8.9  15.3  0.580  0.549  4.6  5.9  0.768  1.2  7.1  8.3  6.8   
4   4.9  0.404  8.3  13.4  0.617  0.614  6.5  7.1  0.919  0.4  6.3  6.7  5.0   

   STL  BLK  TOV   PF   PTS  
0  0.9  

In [3055]:
# Año que deseas agregar
Temporada3 = 2023

# Agregar una nueva columna 'Team' con el valor del año especificado
df3['Temp'] = Temporada3

# Mostrar las primeras filas del DataFrame para verificar
print(df3.head())

              Player Pos  Age     Team  GP  GS    MP    FG   FGA    FG%   3P  \
0      Stephen Curry  PG   34      GSW  56  56  34.7  10.0  20.2  0.493  4.9   
1          John Wall  PG   32      LAC  34   3  22.2   4.1   9.9  0.408  1.0   
2  Russell Westbrook  PG   34  LAL/LAC  73  24  29.1   5.9  13.6  0.436  1.2   
3       LeBron James  PF   38      LAL  55  54  35.5  11.1  22.2  0.500  2.2   
4       Kevin Durant  PF   34  BRK/PHO  47  47  35.6  10.3  18.3  0.560  2.0   

    3PA    3P%   2P   2PA    2P%   eFG%   FT  FTA    FT%  ORB  DRB  TRB  AST  \
0  11.4  0.427  5.1   8.8  0.579  0.614  4.6  5.0  0.915  0.7  5.4  6.1  6.3   
1   3.2  0.303  3.1   6.7  0.459  0.457  2.3  3.3  0.681  0.4  2.3  2.7  5.2   
2   3.9  0.311  4.7   9.7  0.487  0.481  2.8  4.3  0.656  1.2  4.6  5.8  7.5   
3   6.9  0.321  8.9  15.3  0.580  0.549  4.6  5.9  0.768  1.2  7.1  8.3  6.8   
4   4.9  0.404  8.3  13.4  0.617  0.614  6.5  7.1  0.919  0.4  6.3  6.7  5.0   

   STL  BLK  TOV   PF   PTS  Temp  
0 

In [3056]:
np.shape(df3)

(467, 30)

In [3057]:
df3.columns

Index(['Player', 'Pos', 'Age', 'Team', 'GP', 'GS', 'MP', 'FG', 'FGA', 'FG%',
       '3P', '3PA', '3P%', '2P', '2PA', '2P%', 'eFG%', 'FT', 'FTA', 'FT%',
       'ORB', 'DRB', 'TRB', 'AST', 'STL', 'BLK', 'TOV', 'PF', 'PTS', 'Temp'],
      dtype='object')

In [3058]:
# Obtener los valores únicos de la columna team_abbreviation
valores_unicos = df3['Team'].unique()

# Mostrar los valores únicos
print(valores_unicos)

['GSW' 'LAC' 'LAL/LAC' 'LAL' 'BRK/PHO' 'WAS' 'MIL' 'POR' 'BRK/DAL' 'MIN'
 'MIA' 'PHI' 'DAL' 'ATL' 'CHI' 'BRK' 'TOR' 'IND' 'PHO' 'NOP' 'DEN'
 'LAL/MIN' 'CLE' 'OKC' 'CLE/MIA' 'BOS' 'SAC' 'CHO' 'NYK' 'MIN/UTA'
 'HOU/LAC' 'MEM' 'ORL' 'UTA' 'LAL/UTA' 'WAS/TOR' 'LAL/CHI' 'SAS' 'LAC/MEM'
 'NYK/POR' 'NOP/SAS' 'CLE/MEM' 'ORL/PHO' 'DEN/LAC' 'DET' 'LAL/ORL'
 'GSW/DET' 'HOU' 'SAS/TOR' 'BRK/DET' 'LAC/CHO' 'GSW/POR' 'PHI/MIA'
 'LAL/WAS' 'DAL/ATL' 'ORL/IND' 'PHI/POR' 'OKC/PHO' 'MIL/IND' 'BOS/OKC'
 'DET/POR' 'DET/ATL' 'HOU/ATL' 'PHI/PHO' 'DEN/LAL' 'PHI/CHO' 'CHI/IND'
 'BRK/SAC' 'SAS/MIL' 'DET/OKC' 'BOS/TOR' 'BRK/LAC' 'SAS/PHI' 'BRK/MIA'
 'MIA/WAS']


In [3059]:
# Función para obtener el segundo equipo si hay más de uno
def get_second_team(team):
    teams = team.split('/')
    if len(teams) > 1:
        return teams[1]  # Retorna el segundo equipo
    return team  # Retorna el mismo equipo si solo hay uno

# Aplicar la función a la columna 'Team'
df3['Team'] = df3['Team'].apply(get_second_team)

print(df3)

                Player Pos  Age Team  GP  GS    MP    FG   FGA    FG%   3P  \
0        Stephen Curry  PG   34  GSW  56  56  34.7  10.0  20.2  0.493  4.9   
1            John Wall  PG   32  LAC  34   3  22.2   4.1   9.9  0.408  1.0   
2    Russell Westbrook  PG   34  LAC  73  24  29.1   5.9  13.6  0.436  1.2   
3         LeBron James  PF   38  LAL  55  54  35.5  11.1  22.2  0.500  2.2   
4         Kevin Durant  PF   34  PHO  47  47  35.6  10.3  18.3  0.560  2.0   
..                 ...  ..  ...  ...  ..  ..   ...   ...   ...    ...  ...   
462      Justin Minaya  SF   23  POR   4   0  22.3   1.8   5.8  0.304  0.8   
463       Kobi Simmons  SG   25  CHO   5   0   5.6   0.2   1.2  0.167  0.2   
464          Gabe York  SG   29  IND   3   0  18.7   2.7   7.0  0.381  2.0   
465       RaiQuan Gray  PF   23  BRK   1   0  35.0   6.0  12.0  0.500  2.0   
466      Jacob Gilyard  PG   24  MEM   1   0  41.0   1.0   3.0  0.333  1.0   

      3PA    3P%   2P   2PA    2P%   eFG%   FT  FTA    FT%  ORB

In [3060]:
# Obtener los valores únicos de la columna team_abbreviation
valores_unicos = df3['Team'].unique()

# Mostrar los valores únicos
print(valores_unicos)

['GSW' 'LAC' 'LAL' 'PHO' 'WAS' 'MIL' 'POR' 'DAL' 'MIN' 'MIA' 'PHI' 'ATL'
 'CHI' 'BRK' 'TOR' 'IND' 'NOP' 'DEN' 'CLE' 'OKC' 'BOS' 'SAC' 'CHO' 'NYK'
 'UTA' 'MEM' 'ORL' 'SAS' 'DET' 'HOU']


Guardamos el proceso etl de la temporada 2022-2023.

In [3061]:
df3.to_csv("Temporada2022_2023.csv",sep=",")

### ANALISIS SALARIOS 22-23

In [3062]:
df4 = pd.read_csv("nba_2022-23_all_stats_with_salary.csv") 
print(df4.head(11))

    Unnamed: 0            Player Name    Salary Position  Age     Team  GP  \
0            0          Stephen Curry  48070014       PG   34      GSW  56   
1            1              John Wall  47345760       PG   32      LAC  34   
2            2      Russell Westbrook  47080179       PG   34  LAL/LAC  73   
3            3           LeBron James  44474988       PF   38      LAL  55   
4            4           Kevin Durant  44119845       PF   34  BRK/PHO  47   
5            5           Bradley Beal  43279250       SG   29      WAS  50   
6            6          Kawhi Leonard  42492492       SF   31      LAC  52   
7            7            Paul George  42492492       SF   32      LAC  56   
8            8  Giannis Antetokounmpo  42492492       PF   28      MIL  63   
9            9         Damian Lillard  42492492       PG   32      POR  58   
10          10          Klay Thompson  40600080       SF   32      GSW  69   

    GS    MP    FG   FGA    FG%   3P   3PA    3P%    2P   2PA  

In [3063]:
df4.drop(df4.columns[3:51], axis=1, inplace=True)



In [3064]:
df4.drop(df4.columns[0], axis=1, inplace=True)

In [3065]:
df4.drop(df4.columns[2], axis=1, inplace=True)

Lo que hicimos fue quedarnos con las columnas de Salario y nombre de jugador unicamente.

In [3066]:
print(df4.head(11))

              Player Name    Salary
0           Stephen Curry  48070014
1               John Wall  47345760
2       Russell Westbrook  47080179
3            LeBron James  44474988
4            Kevin Durant  44119845
5            Bradley Beal  43279250
6           Kawhi Leonard  42492492
7             Paul George  42492492
8   Giannis Antetokounmpo  42492492
9          Damian Lillard  42492492
10          Klay Thompson  40600080


In [3067]:
ValoresNull4 = df4.isnull().sum() [df4.isnull().sum()>0]
print(ValoresNull4)

Series([], dtype: int64)


Se revisaron valores nulos.

In [3068]:
df4.isnull().sum()

Player Name    0
Salary         0
dtype: int64

In [3069]:
df4.rename(columns={'Player Name': 'Player'}, inplace=True)

In [3070]:
# Año que deseas agregar
Temporada4 = 2023

# Agregar una nueva columna 'Año' con el valor del año especificado
df4['Temp'] = Temporada4

# Mostrar las primeras filas del DataFrame para verificar
print(df4.head())

              Player    Salary  Temp
0      Stephen Curry  48070014  2023
1          John Wall  47345760  2023
2  Russell Westbrook  47080179  2023
3       LeBron James  44474988  2023
4       Kevin Durant  44119845  2023


Se guarda el dataset de salarios ya terminado.

In [3071]:
df4.to_csv("Salarios.csv",sep=",")

# Se comienza a preparar todo para realizar el merge de temporadas.

In [3072]:
df2021_2022 = pd.read_csv("Temporada2021_2022.csv") 
print(df3.head(11))

                   Player Pos  Age Team  GP  GS    MP    FG   FGA    FG%   3P  \
0           Stephen Curry  PG   34  GSW  56  56  34.7  10.0  20.2  0.493  4.9   
1               John Wall  PG   32  LAC  34   3  22.2   4.1   9.9  0.408  1.0   
2       Russell Westbrook  PG   34  LAC  73  24  29.1   5.9  13.6  0.436  1.2   
3            LeBron James  PF   38  LAL  55  54  35.5  11.1  22.2  0.500  2.2   
4            Kevin Durant  PF   34  PHO  47  47  35.6  10.3  18.3  0.560  2.0   
5            Bradley Beal  SG   29  WAS  50  50  33.5   8.9  17.6  0.506  1.6   
6           Kawhi Leonard  SF   31  LAC  52  50  33.6   8.6  16.8  0.512  2.0   
7             Paul George  SF   32  LAC  56  56  34.6   8.2  17.9  0.457  2.8   
8   Giannis Antetokounmpo  PF   28  MIL  63  63  32.1  11.2  20.3  0.553  0.7   
9          Damian Lillard  PG   32  POR  58  58  36.3   9.6  20.7  0.463  4.2   
10          Klay Thompson  SF   32  GSW  69  69  33.0   7.9  18.1  0.436  4.4   

     3PA    3P%    2P   2PA

In [3073]:
# Cargar los tres archivos CSV en DataFrames
df1final = pd.read_csv('Temporada2020_2021.csv')
df2final = pd.read_csv('Temporada2021_2022.csv')
df3final = pd.read_csv('Temporada2022_2023.csv')

# Concatenar los DataFrames
df_combined = pd.concat([df1final, df2final, df3final], ignore_index=True)
df_combined = df_combined.drop('Unnamed: 0', axis=1)
# Guardar el DataFrame combinado en un nuevo archivo CSV
df_combined.to_csv('TempFinal.csv', index=False)

Se realizó el merge y se procede a guardarlo en un .csv

In [3074]:
df_final = pd.read_csv('TempFinal.csv')


In [3075]:
df_final.head(11)

Player Pos  Age Team  GP  GS    MP   FG   FGA    FG%  \
0           Precious Achiuwa  PF   21  MIA  28   2  14.6  2.6   4.4  0.590   
1               Jaylen Adams  PG   24  MIL   6   0   2.8  0.2   1.3  0.125   
2               Steven Adams   C   27  NOP  27  27  28.1  3.5   5.8  0.603   
3                Bam Adebayo   C   23  MIA  26  26  33.6  7.4  12.9  0.573   
4          LaMarcus Aldridge   C   35  SAS  18  18  26.7  5.9  12.5  0.476   
5          Ty-Shon Alexander  SG   22  PHO   3   0   2.7  0.0   1.0  0.000   
6   Nickeil Alexander-Walker  SG   22  NOP  23   3  19.2  3.3   8.2  0.410   
7              Grayson Allen  SG   25  MEM  19   8  23.9  3.2   7.4  0.429   
8              Jarrett Allen   C   22  CLE  28  10  26.2  4.4   6.8  0.642   
9              Jarrett Allen   C   22  BRK  12   5  26.7  3.7   5.4  0.677   
10             Jarrett Allen   C   22  CLE  16   5  25.9  4.9   7.8  0.624   

     3P  3PA    3P%   2P   2PA    2P%   eFG%   FT  FTA    FT%  ORB  DRB   TRB  \
0   0.0  0.0  0.000  2.6   4.4  0.590  0.590  1.3  2.4  0.561  1.3  2.7   4.0   
1   0.0  0.3  0.000  0.2   1.0  0.167  0.125  0.0  0.0  0.000  0.0  0.5   0.5   
2   0.0  0.0  0.000  3.5   5.7  0.606  0.603  1.1  2.3  0.468  4.3  4.6   8.9   
3   0.1  0.2  0.400  7.3  12.7  0.576  0.576  5.1  6.0  0.841  1.9  7.3   9.2   
4   1.3  3.7  0.358  4.6   8.8  0.525  0.529  0.9  1.2  0.762  0.8  3.5   4.3   
5   0.0  0.3  0.000  0.0   0.7  0.000  0.000  0.0  0.0  0.000  0.0  0.3   0.3   
6   1.0  3.8  0.276  2.3   4.4  0.525  0.473  1.1  1.4  0.781  0.2  2.4   2.7   
7   2.3  5.3  0.436  0.8   2.1  0.410  0.586  1.7  1.9  0.892  0.4  2.5   2.9   
8   0.0  0.1  0.250  4.3   6.6  0.651  0.645  3.6  4.7  0.758  2.9  6.1   9.0   
9   0.0  0.0  0.000  3.7   5.4  0.677  0.677  3.8  5.1  0.754  3.2  7.3  10.4   
10  0.1  0.3  0.250  4.8   7.6  0.636  0.628  3.4  4.4  0.761  2.8  5.2   7.9   

    AST  STL  BLK  TOV   PF   PTS  Temp  
0   0.6  0.4  0.5  1.0  1.9   6.5  2021  
1   0.3  0.0  0.0  0.0  0.2   0.3  2021  
2   2.1  1.0  0.6  1.7  1.9   8.0  2021  
3   5.3  1.0  1.0  3.0  2.6  19.9  2021  
4   1.9  0.4  0.9  0.9  1.5  14.1  2021  
5   0.3  0.0  0.0  0.0  0.3   0.0  2021  
6   2.0  1.1  0.3  1.3  1.7   8.8  2021  
7   2.1  1.0  0.2  1.1  1.3  10.4  2021  
8   1.6  0.5  1.6  1.5  1.6  12.3  2021  
9   1.7  0.6  1.6  1.8  1.8  11.2  2021  
10  1.6  0.4  1.7  1.3  1.4  13.2  2021

In [3076]:
# Obtener los valores únicos de la columna team_abbreviation
valores_unicos = df_final['Team'].unique()

# Mostrar los valores únicos
print(valores_unicos)

['MIA' 'MIL' 'NOP' 'SAS' 'PHO' 'MEM' 'CLE' 'BRK' 'ORL' 'LAL' 'POR' 'TOR'
 'CHI' 'WAS' 'UTA' 'SAC' 'CHO' 'NYK' 'DEN' 'LAC' 'GSW' 'OKC' 'MIN' 'DET'
 'DAL' 'IND' 'ATL' 'PHI' 'BOS' 'HOU' 'BKN' 'TOT' 'PHX']


In [3077]:
df_final.columns

Index(['Player', 'Pos', 'Age', 'Team', 'GP', 'GS', 'MP', 'FG', 'FGA', 'FG%',
       '3P', '3PA', '3P%', '2P', '2PA', '2P%', 'eFG%', 'FT', 'FTA', 'FT%',
       'ORB', 'DRB', 'TRB', 'AST', 'STL', 'BLK', 'TOV', 'PF', 'PTS', 'Temp'],
      dtype='object')

In [3078]:
df_final = df_filtered

In [3079]:
df_filtered.head()

Player Pos  Age Team  GP  GS    MP   FG   FGA    FG%   3P  3PA  \
0   Precious Achiuwa   C   22  TOR  73  28  23.6  3.6   8.3  0.439  0.8  2.1   
1       Steven Adams   C   28  MEM  76  75  26.3  2.8   5.1  0.547  0.0  0.0   
2        Bam Adebayo   C   24  MIA  56  56  32.6  7.3  13.0  0.557  0.0  0.1   
3       Santi Aldama  PF   21  MEM  32   0  11.3  1.7   4.1  0.402  0.2  1.5   
4  LaMarcus Aldridge   C   36  BRK  47  12  22.3  5.4   9.7  0.550  0.3  1.0   

     3P%   2P   2PA    2P%   eFG%   FT  FTA    FT%  ORB  DRB   TRB  AST  STL  \
0  0.359  2.9   6.1  0.468  0.486  1.1  1.8  0.595  2.0  4.5   6.5  1.1  0.5   
1  0.000  2.8   5.0  0.548  0.547  1.4  2.6  0.543  4.6  5.4  10.0  3.4  0.9   
2  0.000  7.3  12.9  0.562  0.557  4.6  6.1  0.753  2.4  7.6  10.1  3.4  1.4   
3  0.125  1.5   2.6  0.560  0.424  0.6  1.0  0.625  1.0  1.7   2.7  0.7  0.2   
4  0.304  5.1   8.8  0.578  0.566  1.9  2.2  0.873  1.6  3.9   5.5  0.9  0.3   

   BLK  TOV   PF   PTS  Temp  
0  0.6  1.2  2.1   9.1  2022  
1  0.8  1.5  2.0   6.9  2022  
2  0.8  2.6  3.1  19.1  2022  
3  0.3  0.5  1.1   4.1  2022  
4  1.0  0.9  1.7  12.9  2022

Se le agregan corchetes al principio y al final, dejando los valores de los equipos en el medio, a las columnas Team y TOTAL TEAM.

Esto es para no tener problemas de reconocimiento a la hora de guardar el CSV y enviarlo a SQL, debido a que los datos se separan con '/' en esas columnas.

In [3080]:
df_filtered

Player Pos  Age Team  GP  GS    MP   FG   FGA    FG%   3P  \
0     Precious Achiuwa   C   22  TOR  73  28  23.6  3.6   8.3  0.439  0.8   
1         Steven Adams   C   28  MEM  76  75  26.3  2.8   5.1  0.547  0.0   
2          Bam Adebayo   C   24  MIA  56  56  32.6  7.3  13.0  0.557  0.0   
3         Santi Aldama  PF   21  MEM  32   0  11.3  1.7   4.1  0.402  0.2   
4    LaMarcus Aldridge   C   36  BRK  47  12  22.3  5.4   9.7  0.550  0.3   
..                 ...  ..  ...  ...  ..  ..   ...  ...   ...    ...  ...   
807     Thaddeus Young  PF   33  TOR  26   0  18.3  2.6   5.5  0.465  0.7   
808         Trae Young  PG   23  ATL  76  76  34.9  9.4  20.3  0.460  3.1   
809     Omer Yurtseven   C   23  MIA  56  12  12.6  2.3   4.4  0.526  0.0   
810        Cody Zeller   C   29  POR  27   0  13.1  1.9   3.3  0.567  0.0   
811        Ivica Zubac   C   24  LAC  76  76  24.4  4.1   6.5  0.626  0.0   

     3PA    3P%   2P   2PA    2P%   eFG%   FT  FTA    FT%  ORB  DRB   TRB  \
0    2.1  0.359  2.9   6.1  0.468  0.486  1.1  1.8  0.595  2.0  4.5   6.5   
1    0.0  0.000  2.8   5.0  0.548  0.547  1.4  2.6  0.543  4.6  5.4  10.0   
2    0.1  0.000  7.3  12.9  0.562  0.557  4.6  6.1  0.753  2.4  7.6  10.1   
3    1.5  0.125  1.5   2.6  0.560  0.424  0.6  1.0  0.625  1.0  1.7   2.7   
4    1.0  0.304  5.1   8.8  0.578  0.566  1.9  2.2  0.873  1.6  3.9   5.5   
..   ...    ...  ...   ...    ...    ...  ...  ...    ...  ...  ...   ...   
807  1.7  0.395  1.9   3.9  0.495  0.524  0.5  1.0  0.481  1.5  2.9   4.4   
808  8.0  0.382  6.3  12.3  0.512  0.536  6.6  7.3  0.904  0.7  3.1   3.7   
809  0.2  0.091  2.3   4.2  0.547  0.528  0.7  1.1  0.623  1.5  3.7   5.3   
810  0.1  0.000  1.9   3.2  0.593  0.567  1.4  1.8  0.776  1.9  2.8   4.6   
811  0.0  0.000  4.1   6.5  0.626  0.626  2.2  3.0  0.727  2.9  5.6   8.5   

     AST  STL  BLK  TOV   PF   PTS  Temp  
0    1.1  0.5  0.6  1.2  2.1   9.1  2022  
1    3.4  0.9  0.8  1.5  2.0   6.9  2022  
2    3.4  1.4  0.8  2.6  3.1  19.1  2022  
3    0.7  0.2  0.3  0.5  1.1   4.1  2022  
4    0.9  0.3  1.0  0.9  1.7  12.9  2022  
..   ...  ...  ...  ...  ...   ...   ...  
807  1.7  1.2  0.4  0.8  1.7   6.3  2022  
808  9.7  0.9  0.1  4.0  1.7  28.4  2022  
809  0.9  0.3  0.4  0.7  1.5   5.3  2022  
810  0.8  0.3  0.2  0.7  2.1   5.2  2022  
811  1.6  0.5  1.0  1.5  2.7  10.3  2022  

[812 rows x 30 columns]

In [3081]:
df_filtered.columns

Index(['Player', 'Pos', 'Age', 'Team', 'GP', 'GS', 'MP', 'FG', 'FGA', 'FG%',
       '3P', '3PA', '3P%', '2P', '2PA', '2P%', 'eFG%', 'FT', 'FTA', 'FT%',
       'ORB', 'DRB', 'TRB', 'AST', 'STL', 'BLK', 'TOV', 'PF', 'PTS', 'Temp'],
      dtype='object')

## Ahora creare una player_id para todos los jugadores teniendo en cuenta que hay jugadores que se  repiten. A estos hay que asignarle la misma id cada vez.

In [3082]:
# Obtén los jugadores únicos
unique_players = df_filtered['Player'].unique()

# Crea un diccionario para mapear nombres de jugadores a IDs
player_id_map = {player: idx + 1 for idx, player in enumerate(unique_players)}

# Asigna IDs a los jugadores en el DataFrame
df_filtered['player_id'] = df_filtered['Player'].map(player_id_map)

In [3083]:
np.shape(df_filtered)

(812, 31)

In [3084]:
df_filtered.isnull().sum()

Player       0
Pos          0
Age          0
Team         0
GP           0
GS           0
MP           0
FG           0
FGA          0
FG%          0
3P           0
3PA          0
3P%          0
2P           0
2PA          0
2P%          0
eFG%         0
FT           0
FTA          0
FT%          0
ORB          0
DRB          0
TRB          0
AST          0
STL          0
BLK          0
TOV          0
PF           0
PTS          0
Temp         0
player_id    0
dtype: int64

In [3085]:
df_filtered.columns

Index(['Player', 'Pos', 'Age', 'Team', 'GP', 'GS', 'MP', 'FG', 'FGA', 'FG%',
       '3P', '3PA', '3P%', '2P', '2PA', '2P%', 'eFG%', 'FT', 'FTA', 'FT%',
       'ORB', 'DRB', 'TRB', 'AST', 'STL', 'BLK', 'TOV', 'PF', 'PTS', 'Temp',
       'player_id'],
      dtype='object')

In [3086]:
# Agrupamos por 'player_id' y verificamos si cada grupo tiene un solo nombre único en la columna 'Player'
validation = df_filtered.groupby('player_id')['Player'].nunique()

# Filtramos para encontrar cualquier 'player_id' que tenga más de un nombre asociado
invalid_ids = validation[validation > 1]

if invalid_ids.empty:
    print("¡Éxito! Todos los player_id tienen un único nombre de jugador asignado.")
else:
    print("Se encontraron inconsistencias en los siguientes player_id:")
    print(invalid_ids)

¡Éxito! Todos los player_id tienen un único nombre de jugador asignado.


In [3087]:
df_filtered.isnull().sum()

Player       0
Pos          0
Age          0
Team         0
GP           0
GS           0
MP           0
FG           0
FGA          0
FG%          0
3P           0
3PA          0
3P%          0
2P           0
2PA          0
2P%          0
eFG%         0
FT           0
FTA          0
FT%          0
ORB          0
DRB          0
TRB          0
AST          0
STL          0
BLK          0
TOV          0
PF           0
PTS          0
Temp         0
player_id    0
dtype: int64

In [3088]:
df_filtered.head()

Player Pos  Age Team  GP  GS    MP   FG   FGA    FG%   3P  3PA  \
0   Precious Achiuwa   C   22  TOR  73  28  23.6  3.6   8.3  0.439  0.8  2.1   
1       Steven Adams   C   28  MEM  76  75  26.3  2.8   5.1  0.547  0.0  0.0   
2        Bam Adebayo   C   24  MIA  56  56  32.6  7.3  13.0  0.557  0.0  0.1   
3       Santi Aldama  PF   21  MEM  32   0  11.3  1.7   4.1  0.402  0.2  1.5   
4  LaMarcus Aldridge   C   36  BRK  47  12  22.3  5.4   9.7  0.550  0.3  1.0   

     3P%   2P   2PA    2P%   eFG%   FT  FTA    FT%  ORB  DRB   TRB  AST  STL  \
0  0.359  2.9   6.1  0.468  0.486  1.1  1.8  0.595  2.0  4.5   6.5  1.1  0.5   
1  0.000  2.8   5.0  0.548  0.547  1.4  2.6  0.543  4.6  5.4  10.0  3.4  0.9   
2  0.000  7.3  12.9  0.562  0.557  4.6  6.1  0.753  2.4  7.6  10.1  3.4  1.4   
3  0.125  1.5   2.6  0.560  0.424  0.6  1.0  0.625  1.0  1.7   2.7  0.7  0.2   
4  0.304  5.1   8.8  0.578  0.566  1.9  2.2  0.873  1.6  3.9   5.5  0.9  0.3   

   BLK  TOV   PF   PTS  Temp  player_id  
0  0.6  1.2  2.1   9.1  2022          1  
1  0.8  1.5  2.0   6.9  2022          2  
2  0.8  2.6  3.1  19.1  2022          3  
3  0.3  0.5  1.1   4.1  2022          4  
4  1.0  0.9  1.7  12.9  2022          5

Se modifica el formato de las ids.

In [3089]:
df_filtered['player_id'] = df_filtered['player_id'].astype('int')

### Se crea una llave primaria compuesta unificada con la id de jugadores y la temporada en la que jugó.

Esto se decidió para que la tabla de hechos tenga todas ids unicas.

In [3090]:
# Crear una nueva columna 'primary_key' que combine 'id' y 'Temp'
df_filtered['primary_key_composed'] = df_filtered['player_id'].astype(str) + "_" + df_filtered['Temp'].astype(str)

# Verificar las primeras filas de la tabla con la nueva columna
print(df_filtered[['player_id', 'Temp', 'primary_key_composed']].head())

   player_id  Temp primary_key_composed
0          1  2022               1_2022
1          2  2022               2_2022
2          3  2022               3_2022
3          4  2022               4_2022
4          5  2022               5_2022


In [3091]:
# Verificar si hay duplicados en la columna 'id'
duplicated_ids = df_filtered[df_filtered['primary_key_composed'].duplicated(keep=False)]

# Mostrar los duplicados encontrados (si existen)
if not duplicated_ids.empty:
    print("Se encontraron los siguientes duplicados en la columna 'primary_key_composed':")
    print(duplicated_ids)
else:
    print("No se encontraron duplicados en la columna 'primary_key_composed'.")



Se encontraron los siguientes duplicados en la columna 'primary_key_composed':
                       Player Pos  Age Team  GP  GS    MP   FG   FGA    FG%  \
5    Nickeil Alexander-Walker  SG   23  NOP  65  21  22.6  3.9  10.5  0.372   
6    Nickeil Alexander-Walker  SG   23  NOP  50  19  26.3  4.7  12.6  0.375   
7    Nickeil Alexander-Walker  SG   23  UTA  15   2   9.9  1.1   3.2  0.333   
11            Justin Anderson  SF   28  CLE  16   6  19.8  2.3   5.9  0.379   
12            Justin Anderson  SF   28  CLE   3   0  15.7  1.3   2.7  0.500   
..                        ...  ..  ...  ...  ..  ..   ...  ...   ...    ...   
802              Moses Wright  PF   23  LAC   1   0   1.0  0.0   0.0  0.000   
803              Moses Wright  PF   23  DAL   3   0   4.3  0.3   1.3  0.250   
805            Thaddeus Young  PF   33  SAS  52   1  16.3  2.7   5.2  0.518   
806            Thaddeus Young  PF   33  SAS  26   1  14.2  2.8   4.9  0.578   
807            Thaddeus Young  PF   33  TOR  26   0 

Verificamos el exito de nuestro codigo, observando los duplicados.

In [3092]:
# Verificar si hay duplicados en la columna 'id'
duplicated_ids = df_filtered[df_filtered['primary_key_composed'].duplicated(keep=False)]

# Mostrar los duplicados encontrados (si existen)
if not duplicated_ids.empty:
    print("Se encontraron los siguientes duplicados en la columna 'primary_key_composed':")
    print(duplicated_ids)
else:
    print("No se encontraron duplicados en la columna 'primary_key_composed'.")


Se encontraron los siguientes duplicados en la columna 'primary_key_composed':
                       Player Pos  Age Team  GP  GS    MP   FG   FGA    FG%  \
5    Nickeil Alexander-Walker  SG   23  NOP  65  21  22.6  3.9  10.5  0.372   
6    Nickeil Alexander-Walker  SG   23  NOP  50  19  26.3  4.7  12.6  0.375   
7    Nickeil Alexander-Walker  SG   23  UTA  15   2   9.9  1.1   3.2  0.333   
11            Justin Anderson  SF   28  CLE  16   6  19.8  2.3   5.9  0.379   
12            Justin Anderson  SF   28  CLE   3   0  15.7  1.3   2.7  0.500   
..                        ...  ..  ...  ...  ..  ..   ...  ...   ...    ...   
802              Moses Wright  PF   23  LAC   1   0   1.0  0.0   0.0  0.000   
803              Moses Wright  PF   23  DAL   3   0   4.3  0.3   1.3  0.250   
805            Thaddeus Young  PF   33  SAS  52   1  16.3  2.7   5.2  0.518   
806            Thaddeus Young  PF   33  SAS  26   1  14.2  2.8   4.9  0.578   
807            Thaddeus Young  PF   33  TOR  26   0 

In [3093]:
df_filtered

Player Pos  Age Team  GP  GS    MP   FG   FGA    FG%   3P  \
0     Precious Achiuwa   C   22  TOR  73  28  23.6  3.6   8.3  0.439  0.8   
1         Steven Adams   C   28  MEM  76  75  26.3  2.8   5.1  0.547  0.0   
2          Bam Adebayo   C   24  MIA  56  56  32.6  7.3  13.0  0.557  0.0   
3         Santi Aldama  PF   21  MEM  32   0  11.3  1.7   4.1  0.402  0.2   
4    LaMarcus Aldridge   C   36  BRK  47  12  22.3  5.4   9.7  0.550  0.3   
..                 ...  ..  ...  ...  ..  ..   ...  ...   ...    ...  ...   
807     Thaddeus Young  PF   33  TOR  26   0  18.3  2.6   5.5  0.465  0.7   
808         Trae Young  PG   23  ATL  76  76  34.9  9.4  20.3  0.460  3.1   
809     Omer Yurtseven   C   23  MIA  56  12  12.6  2.3   4.4  0.526  0.0   
810        Cody Zeller   C   29  POR  27   0  13.1  1.9   3.3  0.567  0.0   
811        Ivica Zubac   C   24  LAC  76  76  24.4  4.1   6.5  0.626  0.0   

     3PA    3P%   2P   2PA    2P%   eFG%   FT  FTA    FT%  ORB  DRB   TRB  \
0    2.1  0.359  2.9   6.1  0.468  0.486  1.1  1.8  0.595  2.0  4.5   6.5   
1    0.0  0.000  2.8   5.0  0.548  0.547  1.4  2.6  0.543  4.6  5.4  10.0   
2    0.1  0.000  7.3  12.9  0.562  0.557  4.6  6.1  0.753  2.4  7.6  10.1   
3    1.5  0.125  1.5   2.6  0.560  0.424  0.6  1.0  0.625  1.0  1.7   2.7   
4    1.0  0.304  5.1   8.8  0.578  0.566  1.9  2.2  0.873  1.6  3.9   5.5   
..   ...    ...  ...   ...    ...    ...  ...  ...    ...  ...  ...   ...   
807  1.7  0.395  1.9   3.9  0.495  0.524  0.5  1.0  0.481  1.5  2.9   4.4   
808  8.0  0.382  6.3  12.3  0.512  0.536  6.6  7.3  0.904  0.7  3.1   3.7   
809  0.2  0.091  2.3   4.2  0.547  0.528  0.7  1.1  0.623  1.5  3.7   5.3   
810  0.1  0.000  1.9   3.2  0.593  0.567  1.4  1.8  0.776  1.9  2.8   4.6   
811  0.0  0.000  4.1   6.5  0.626  0.626  2.2  3.0  0.727  2.9  5.6   8.5   

     AST  STL  BLK  TOV   PF   PTS  Temp  player_id primary_key_composed  
0    1.1  0.5  0.6  1.2  2.1   9.1  2022          1               1_2022  
1    3.4  0.9  0.8  1.5  2.0   6.9  2022          2               2_2022  
2    3.4  1.4  0.8  2.6  3.1  19.1  2022          3               3_2022  
3    0.7  0.2  0.3  0.5  1.1   4.1  2022          4               4_2022  
4    0.9  0.3  1.0  0.9  1.7  12.9  2022          5               5_2022  
..   ...  ...  ...  ...  ...   ...   ...        ...                  ...  
807  1.7  1.2  0.4  0.8  1.7   6.3  2022        601             601_2022  
808  9.7  0.9  0.1  4.0  1.7  28.4  2022        602             602_2022  
809  0.9  0.3  0.4  0.7  1.5   5.3  2022        603             603_2022  
810  0.8  0.3  0.2  0.7  2.1   5.2  2022        604             604_2022  
811  1.6  0.5  1.0  1.5  2.7  10.3  2022        605             605_2022  

[812 rows x 32 columns]

Reordenamos las columnas en base a una estandarización propuesta por el equipo encargado de la arquitectura.

In [3094]:
df_filtered = df_filtered[['primary_key_composed', 'player_id', 'Player', 'Pos', 'Age', 'Team', 'GP', 'GS', 'MP', 'FG', 'FGA', 'FG%',
            '3P', '3PA', '3P%', '2P', '2PA', '2P%', 'eFG%', 'FT', 'FTA', 'FT%',
            'ORB', 'DRB', 'TRB', 'AST', 'STL', 'BLK', 'TOV', 'PF', 'PTS', 'Temp']]
            

In [3095]:
df_filtered = df_filtered.rename(columns={'Temp': 'season_id'})
df_filtered = df_filtered.rename(columns={'Team': 'team_abbreviation'})
df_filtered = df_filtered.rename(columns={'Player': 'player_name'})
df_filtered = df_filtered.rename(columns={'Pos': 'position'})
df_filtered = df_filtered.rename(columns={'Age': 'age'})

In [3096]:
print(df_filtered.dtypes)

primary_key_composed     object
player_id                 int64
player_name              object
position                 object
age                       int64
team_abbreviation        object
GP                        int64
GS                        int64
MP                      float64
FG                      float64
FGA                     float64
FG%                     float64
3P                      float64
3PA                     float64
3P%                     float64
2P                      float64
2PA                     float64
2P%                     float64
eFG%                    float64
FT                      float64
FTA                     float64
FT%                     float64
ORB                     float64
DRB                     float64
TRB                     float64
AST                     float64
STL                     float64
BLK                     float64
TOV                     float64
PF                      float64
PTS                     float64
season_i

In [3097]:
np.shape(df_filtered)

(812, 32)

In [3098]:
# Filtrar filas donde 'Team' es 'TOT'
df_tot = df_filtered[df_filtered['team_abbreviation'] == 'TOT']

# Mostrar el resultado
print(df_tot)

    primary_key_composed  player_id         player_name position  age  \
54               45_2022         45       D?vis Bert?ns       PF   29   
189             149_2022        149        Goran Dragi?       PG   35   
601             455_2022        455  Kristaps Porzi??is       PF   26   
651             490_2022        490    TomᚠSatoransk?       SG   30   

    team_abbreviation  GP  GS    MP   FG   FGA    FG%   3P  3PA    3P%   2P  \
54                TOT  56   0  14.4  1.8   4.9  0.360  1.4  4.1  0.335  0.4   
189               TOT  21   8  23.7  2.7   7.2  0.377  0.8  3.0  0.254  2.0   
601               TOT  51  51  29.0  6.9  15.0  0.459  1.5  4.9  0.310  5.4   
651               TOT  55  13  16.5  1.3   3.5  0.377  0.2  1.0  0.208  1.1   

      2PA    2P%   eFG%   FT  FTA    FT%  ORB  DRB  TRB  AST  STL  BLK  TOV  \
54    0.8  0.489  0.500  0.6  0.7  0.900  0.2  1.9  2.1  0.6  0.3  0.2  0.3   
189   4.2  0.466  0.430  1.3  1.6  0.818  0.6  2.5  3.1  4.1  0.9  0.2  1.6   
6

In [3099]:
# Obtener los valores únicos de la columna team_abbreviation
valores_unicos = df_tot['player_name'].unique()

# Mostrar los valores únicos
print(valores_unicos)

['D?vis Bert?ns' 'Goran Dragi?' 'Kristaps Porzi??is' 'Tomá\x9a Satoransk?']


Davis Bertans - WAS
Goran Dragić - TOR
Kristaps Porzingis - DAL (se unió a WAS en un trade durante la temporada)
Tomáš Satoranský - NOP

In [3100]:
# Mapeo de jugadores a equipos
player_to_team = {
    'D?vis Bert?ns': 'WAS',
    'Goran Dragi?': 'TOR',
    'Kristaps Porzi??is': 'WAS',
    'Tomá\x9a Satoransk?': 'NOP',
}

# Reemplazar 'TOT' por el equipo correspondiente
def replace_team(row):
    if 'TOT' in row['team_abbreviation']:
        return player_to_team.get(row['player_name'], row['team_abbreviation'])  # Reemplaza si hay un mapeo, de lo contrario mantiene el valor original
    return row['team_abbreviation']

df_filtered['team_abbreviation'] = df_filtered.apply(replace_team, axis=1)

print(df_filtered)

    primary_key_composed  player_id        player_name position  age  \
0                 1_2022          1   Precious Achiuwa        C   22   
1                 2_2022          2       Steven Adams        C   28   
2                 3_2022          3        Bam Adebayo        C   24   
3                 4_2022          4       Santi Aldama       PF   21   
4                 5_2022          5  LaMarcus Aldridge        C   36   
..                   ...        ...                ...      ...  ...   
807             601_2022        601     Thaddeus Young       PF   33   
808             602_2022        602         Trae Young       PG   23   
809             603_2022        603     Omer Yurtseven        C   23   
810             604_2022        604        Cody Zeller        C   29   
811             605_2022        605        Ivica Zubac        C   24   

    team_abbreviation  GP  GS    MP   FG   FGA    FG%   3P  3PA    3P%   2P  \
0                 TOR  73  28  23.6  3.6   8.3  0.439  0

In [3101]:
# Filtrar filas donde 'Team' es 'TOT'
df_tot = df_filtered[df_filtered['team_abbreviation'] == 'TOT']

# Mostrar el resultado
print(df_tot)

Empty DataFrame
Columns: [primary_key_composed, player_id, player_name, position, age, team_abbreviation, GP, GS, MP, FG, FGA, FG%, 3P, 3PA, 3P%, 2P, 2PA, 2P%, eFG%, FT, FTA, FT%, ORB, DRB, TRB, AST, STL, BLK, TOV, PF, PTS, season_id]
Index: []


Se guarda la tabla de hechos final.

In [3102]:
df_filtered.to_csv('player_final_merged_with_stats.csv', index=False)